<!doctype html>



  


<html class="theme-next muse use-motion" lang="en">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>



<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />












  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  
    
      
    

    
  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Arial, Lato:300,300italic,400,400italic,700,700italic|monospace, consolas, Courier New:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.0" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="CGmapTools, DNA methylation, Bisulfite sequencing, Allele specific methylation, SNP calling" />













<meta name="description" content="Advanced Features1. Unified data formats
Use standard CGmap/ATCGmap formats as central format, for ease of sharing methylomes. ATCGmap/CGmap could be derived from BAM files, which are produced by BS-S">
<meta property="og:type" content="website">
<meta property="og:title" content="Command-line Toolset for Bisulfite Sequencing Data Analysis">
<meta property="og:url" content="https://cgmaptools.github.io/index.html">
<meta property="og:site_name" content="CGmapTools">
<meta property="og:description" content="Advanced Features1. Unified data formats
Use standard CGmap/ATCGmap formats as central format, for ease of sharing methylomes. ATCGmap/CGmap could be derived from BAM files, which are produced by BS-S">
<meta property="og:updated_time" content="2017-02-19T01:37:36.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Command-line Toolset for Bisulfite Sequencing Data Analysis">
<meta name="twitter:description" content="Advanced Features1. Unified data formats
Use standard CGmap/ATCGmap formats as central format, for ease of sharing methylomes. ATCGmap/CGmap could be derived from BAM files, which are produced by BS-S">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Muse',
    sidebar: {"position":"left","display":"remove"},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: 'Author'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="https://cgmaptools.github.io/"/>





  <title>
  

  
    Command-line Toolset for Bisulfite Sequencing Data Analysis | CGmapTools
  
</title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="en">

  










  
  
    
  

  <div class="container one-collumn sidebar-position-left  ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner">
<div class="site-meta custom-logo">
  
    <div class="site-meta-headline">
      <a>
        <img class="custom-logo-image" src="/images/CGmapTools_logo.png"
             alt="CGmapTools"/>
      </a>
    </div>
  

  <div class="custom-logo-site-title">
    <a href="/"  class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <span class="site-title">CGmapTools</span>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>
    
      <p class="site-subtitle"></p>
    
</div>

<div class="site-nav-toggle">
  <button>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
  </button>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            Home
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tutorial">
          <a href="/cgmaptools_documentation/what-is-cgmaptools.html" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-book"></i> <br />
            
            Tutorial
          </a>
        </li>
      
        
        <li class="menu-item menu-item-source code">
          <a href="/source-code" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-code"></i> <br />
            
            Source Code
          </a>
        </li>
      
        
        <li class="menu-item menu-item-quick start">
          <a href="/quick-start" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-play"></i> <br />
            
            Quick Start
          </a>
        </li>
      
        
        <li class="menu-item menu-item-release log">
          <a href="/release-log" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-list-ol"></i> <br />
            
            Release log
          </a>
        </li>
      
        
        <li class="menu-item menu-item-q&a">
          <a href="/q-a" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-question-circle"></i> <br />
            
            Q&A
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
  <header class="post-header">

	<h2 class="post-title" itemprop="name headline">Command-line Toolset for Bisulfite Sequencing Data Analysis</h2>



</header>

    
    
      <h1 id="Advanced-Features"><a href="#Advanced-Features" class="headerlink" title="Advanced Features"></a>Advanced Features</h1><h3 id="1-Unified-data-formats"><a href="#1-Unified-data-formats" class="headerlink" title="1. Unified data formats"></a>1. Unified data formats</h3><blockquote>
<p>Use standard <a href="https://github.com/BSSeeker/BSseeker2#output-files" target="_blank" rel="external">CGmap/ATCGmap</a> formats as central format, for ease of sharing methylomes. ATCGmap/CGmap could be derived from BAM files, which are produced by <a href="http://pellegrini.mcdb.ucla.edu/BS_Seeker2/" target="_blank" rel="external">BS-Seeker2</a> or <a href="http://www.bioinformatics.babraham.ac.uk/projects/bismark/" target="_blank" rel="external">Bismark</a></p>
</blockquote>
<h3 id="2-Command-lines-convenient-for-parallel-processing-and-to-be-extended"><a href="#2-Command-lines-convenient-for-parallel-processing-and-to-be-extended" class="headerlink" title="2. Command-lines convenient for parallel processing and to be extended"></a>2. Command-lines convenient for parallel processing and to be extended</h3><blockquote>
<p>Support both Linux and MAC; Support STDIN and STDOUT, so as to support pipe</p>
</blockquote>
<h3 id="3-Support-instantly-retrieving-by-designed-binary-file-formats"><a href="#3-Support-instantly-retrieving-by-designed-binary-file-formats" class="headerlink" title="3. Support instantly retrieving by designed binary file formats"></a>3. Support instantly retrieving by designed binary file formats</h3><blockquote>
<p>binary format CGbz for CGmap, ATCGbz for ATCGmap</p>
</blockquote>
<h4 id="4-Use-dynamic-fragmentation-strategy-for-DMR-analysis"><a href="#4-Use-dynamic-fragmentation-strategy-for-DMR-analysis" class="headerlink" title="4. Use dynamic fragmentation strategy for DMR analysis"></a>4. Use dynamic fragmentation strategy for DMR analysis</h4><blockquote>
<p>useful for low-coverage (WGBS) and fragmented (RRBS) DNA methylomes</p>
</blockquote>
<h4 id="5-Novel-strategies-for-SNV-calling"><a href="#5-Novel-strategies-for-SNV-calling" class="headerlink" title="5. Novel strategies for SNV calling"></a>5. Novel strategies for SNV calling</h4><blockquote>
<p>introduced ambigous genotype prediction, and significantly improve the precision especially in heterozygous SNVs</p>
</blockquote>
<h4 id="6-Support-ASM-analyses"><a href="#6-Support-ASM-analyses" class="headerlink" title="6. Support ASM analyses"></a>6. Support ASM analyses</h4><blockquote>
<p>use precise heterozygous SNVs as input, designed Tanghulu plots for showing allele-specific DNA methylation on reads</p>
</blockquote>
<h4 id="7-Advanced-methylation-analysis-and-visualization"><a href="#7-Advanced-methylation-analysis-and-visualization" class="headerlink" title="7. Advanced methylation analysis and visualization"></a>7. Advanced methylation analysis and visualization</h4><blockquote>
<p>provide user-friendly functions for visualizing methylomes at multiple levels, such as design a Tanghulu plot for visualizing methylation status on original reads, and design a Lollipop plot to reveal both lowly-covered cytosines and un-methylated cytosines in a local region</p>
</blockquote>
<h1 id="Example"><a href="#Example" class="headerlink" title="Example"></a>Example</h1><pre><code>$ cgmaptools -h
Program : cgmaptools (Tools for analysis in CGmap/ATCGmap format)
Version:  0.0.1
Usage:    cgmaptools &lt;command&gt; [options]
Commands:
  -- File manipulation
     convert     + data format conversion tools
     fetch       + fetch a region by random accessing
     refill        refill the missing columns
     intersect     intersect two files
     merge2      + merge two files into one
     mergelist   + merge a list of files
     sort          sort lines by chromosome and position
     split       + split file by chromosomes
     select      + select lines by region/site
  -- SNV analysis
     snv           snv analysis
  -- Methylation analysis
     dms           differentially methylated site analysis
     dmr           differentially methylated region analysis
     asm           allele-specific methylation analysis
     mbed          average methylation level in regions
     mbin        * single sample, mC levels in bins
     mmbin         multiple samples, mC levels in bins
     mfg           methlation levels across fragmented region
     mstat       * methyaltion statistic
     mtr           methylation level to each region
  -- Coverage analysis
     oac        +* overall coverage (for ATCGmap)
     mec        +* methylation effective coverage (for CGmap)
  -- Graph related functions
     lollipop    * show local mC levels as lollipop bars
     heatmap     * global mC distribution for multiple samples
     fragreg     * show mC profile across fragmented regions
     tanghulu    * show local mapped reads in Tanghulu shape
  -- Other Utils
     findCCGG      get MspI cutting sites for RRBS
     bed2fragreg   get fragmented region based on region
Note:
  Commands support figures generation are marked with &quot;*&quot;
  Commands contain sub-commands are marked with &quot;+&quot;
Authors:
  GUO, Weilong; guoweilong@126.com;  http://guoweilong.github.io
  ZHU, Ping; pingzhu.work@gmail.com; http://perry-zhu.github.io
</code></pre>
    
  </div>

          
          </div>
          


          
  <div class="comments" id="comments">
    
  </div>


        </div>
        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2017</span>
  <span class="with-love">
    <i class="fa fa-"></i>
  </span>
  <span class="author" itemprop="copyrightHolder"></span>
</div>


<div class="powered-by">
  Powered by <a class="theme-link" href="https://hexo.io">Hexo</a>
</div>

<div class="theme-info">
  Theme -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Muse
  </a>
</div>


        

        
      </div>
    </footer>

    <div class="back-to-top">
      <i class="fa fa-arrow-up"></i>
    </div>
  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  




  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.0"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.0"></script>



  
  

  

  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.0"></script>



  



  




	





  





  

  
      <!-- UY BEGIN -->
      <script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid="></script>
      <!-- UY END -->
  




  
  

  

  

  

  


</body>
</html>
