<!doctype html>



  


<html class="theme-next muse use-motion" lang="en">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>



<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />












  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  
    
      
    

    
  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Arial, Lato:300,300italic,400,400italic,700,700italic|monospace, consolas, Courier New:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.0" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="CGmapTools, DNA methylation, Bisulfite sequencing, Allele specific methylation, SNP calling" />













<meta name="description" content="In this section, we only show you some basic implementation of the CGmapTools toolset, please go to the tutorial page for more detail introduction of our CGmapTools.
Installation1234567891011121314151">
<meta property="og:type" content="website">
<meta property="og:title" content="Quick Start">
<meta property="og:url" content="https://cgmaptools.github.io/quick-start/index.html">
<meta property="og:site_name" content="CGmapTools">
<meta property="og:description" content="In this section, we only show you some basic implementation of the CGmapTools toolset, please go to the tutorial page for more detail introduction of our CGmapTools.
Installation1234567891011121314151">
<meta property="og:updated_time" content="2017-02-19T01:41:53.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Quick Start">
<meta name="twitter:description" content="In this section, we only show you some basic implementation of the CGmapTools toolset, please go to the tutorial page for more detail introduction of our CGmapTools.
Installation1234567891011121314151">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Muse',
    sidebar: {"position":"left","display":"remove"},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: 'Author'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="https://cgmaptools.github.io/quick-start/"/>





  <title>
  

  
    Quick Start | CGmapTools
  
</title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="en">

  










  
  
    
  

  <div class="container one-collumn sidebar-position-left  ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner">
<div class="site-meta custom-logo">
  
    <div class="site-meta-headline">
      <a>
        <img class="custom-logo-image" src="/images/CGmapTools_logo.png"
             alt="CGmapTools"/>
      </a>
    </div>
  

  <div class="custom-logo-site-title">
    <a href="/"  class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <span class="site-title">CGmapTools</span>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>
    
      <p class="site-subtitle">(40 functions, 13k lines of codes)</p>
    
</div>

<div class="site-nav-toggle">
  <button>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
  </button>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            Home
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tutorial">
          <a href="/cgmaptools_documentation/what-is-cgmaptools.html" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-book"></i> <br />
            
            Tutorial
          </a>
        </li>
      
        
        <li class="menu-item menu-item-source code">
          <a href="/source-code" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-code"></i> <br />
            
            Source Code
          </a>
        </li>
      
        
        <li class="menu-item menu-item-release log">
          <a href="/release-log" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-list-ol"></i> <br />
            
            Release log
          </a>
        </li>
      
        
        <li class="menu-item menu-item-quick start">
          <a href="/quick-start" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-play"></i> <br />
            
            Quick Start
          </a>
        </li>
      
        
        <li class="menu-item menu-item-how to">
          <a href="/how-to" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            menu.how to
          </a>
        </li>
      
        
        <li class="menu-item menu-item-q&a">
          <a href="/q-a" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-question-circle"></i> <br />
            
            Q&A
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
  <header class="post-header">

	<h2 class="post-title" itemprop="name headline">Quick Start</h2>



</header>

    
    
      <p>In this section, we only show you some basic implementation of the CGmapTools toolset, please go to the tutorial page for more detail introduction of our CGmapTools.</p>
<h2 id="Installation"><a href="#Installation" class="headerlink" title="Installation"></a>Installation</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># decompress the gzip tarball</span></div><div class="line">tar -zxf cgmaptools.1.0.0.tar.gz</div><div class="line"><span class="comment"># go to cgmaptool.1.0.0</span></div><div class="line"><span class="built_in">cd</span> cgmaptools.1.0.0</div><div class="line"><span class="comment"># install</span></div><div class="line">sh ./install.sh</div><div class="line"></div><div class="line"><span class="comment"># add cgmaptools to your system PATH if necessary</span></div><div class="line"><span class="comment"># open ~/.bashrc</span></div><div class="line">vi ~/.bashrc</div><div class="line"><span class="comment"># add the following lines to the end of ~/.bashrc</span></div><div class="line"><span class="built_in">export</span> PATH=/path/to/cgmaptools/:<span class="variable">$PATH</span></div><div class="line"><span class="comment"># if you want to use bin-command directly, add the following lines to the end of ~/.bashrc</span></div><div class="line"><span class="built_in">export</span> PATH=/path/to/cgmaptools/bin/:<span class="variable">$PATH</span></div><div class="line"><span class="comment"># then, souce your ~/.bashrc</span></div><div class="line"><span class="built_in">source</span> ~/.bashrc</div></pre></td></tr></table></figure>
<h2 id="1-DNA-methylation-profiles-from-alignments"><a href="#1-DNA-methylation-profiles-from-alignments" class="headerlink" title="1. DNA methylation profiles from alignments"></a>1. DNA methylation profiles from alignments</h2><p>Suppose we have aligned reads stored in BAM format files. Currently, we support alignments in bam format from BSseeker2 and bismark.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># Obtain CGmap and ATCGmap</span></div><div class="line">cgmaptools convert bam2cgmap -b &lt;BAM&gt; -g &lt;genome.fa&gt; -o &lt;prefix&gt;</div></pre></td></tr></table></figure>
<blockquote>
<p>If itâ€™s paired-end sequencing strategy, we recommend you add <code>--rmOverlap</code> to the above command to remove overlapped region of two mate reads avoiding twice consideration of the same DNA fragment.</p>
</blockquote>
<h2 id="2-SNP-calling-using-bisulfite-sequencing-data"><a href="#2-SNP-calling-using-bisulfite-sequencing-data" class="headerlink" title="2. SNP calling using bisulfite sequencing data"></a>2. SNP calling using bisulfite sequencing data</h2><p>After we have <em>ATCGmap</em> file, SNP calling can be performed.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># using bayes model</span></div><div class="line">cgmaptools snv -m bayes --bayes-dynamicP -i &lt;ATCGmap&gt; -v &lt;VCF&gt;</div></pre></td></tr></table></figure>
<blockquote>
<p>There are two strategies we can use to call SNPs, binomial or bayesian mode. The default choice is the binomial mode as itâ€™s faster, but the overall prediction precision of bayesian mode is better. If you are not in hurry, we recommand you the beyasian mode for more accuracy callings.<br>Also to increase prediction precision, <code>--bayes-dynamicP</code> is recommanded when using bayesian mode, which dynamicsly adjust p-value threshoulds according to the read coverage.</p>
</blockquote>
<h2 id="3-Allele-specific-methylation-ASM-analysis"><a href="#3-Allele-specific-methylation-ASM-analysis" class="headerlink" title="3. Allele-specific methylation (ASM) analysis"></a>3. Allele-specific methylation (ASM) analysis</h2><p>ASM is easy to perform by using <code>asm</code> utility in CGmapTools if you have already got SNP sites in <em>vcf</em> format.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># perform allele specific methylation</span></div><div class="line">cgmaptools asm [options] -r &lt;ref.fa&gt; -b &lt;input.bam&gt; <span class="_">-l</span> &lt;snp.vcf&gt;</div></pre></td></tr></table></figure>
<h2 id="4-Visualization-of-ASM"><a href="#4-Visualization-of-ASM" class="headerlink" title="4. Visualization of ASM"></a>4. Visualization of ASM</h2><p>We present a new way to automatically display methylation on mapped reads, which we call <strong>Tanghulu</strong> plot.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># display reads mapped in a specified region or linked by a given SNP</span></div><div class="line">cgmaptools tanghulu -r &lt;genome.fa&gt; -b &lt;BAM&gt; <span class="_">-l</span> &lt;region&gt;</div></pre></td></tr></table></figure>
<p><strong>Tanghulu</strong> plot supports two kinds of region format.</p>
<p>format 1, e.g.: <u>chr1:133</u>. This will ask CGmapTools to show reads mapped over this site. If this is a heterozygous SNP site, we recommand you to add assign the genotype option <code>-g</code> to resolve more vague reads caused by bisulfite coversion.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># display reads linked by a given SNP</span></div><div class="line">cgmaptools tanghulu -r genome.fa -b BAM <span class="_">-l</span> chr1:133 -g GT</div></pre></td></tr></table></figure>
<p>format 2, e.g.: <u>chr1:133-144</u>. This region format will tell CGmapTools to show reads mapped over this region.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="comment"># display reads mapped over a specified region</span></div><div class="line">cgmaptools tanghulu -r genome.fa -b BAM <span class="_">-l</span> chr1:133-144</div></pre></td></tr></table></figure>

    
  </div>

          
          </div>
          


          
  <div class="comments" id="comments">
    
  </div>


        </div>
        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2017</span>
  <span class="with-love">
    <i class="fa fa-"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Weilong Guo and Ping Zhu</span>
</div>


<div class="powered-by">
  Powered by <a class="theme-link" href="https://hexo.io">Hexo</a>
</div>

<div class="theme-info">
  Theme -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Muse
  </a>
</div>


        

        
      </div>
    </footer>

    <div class="back-to-top">
      <i class="fa fa-arrow-up"></i>
    </div>
  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  




  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.0"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.0"></script>



  
  

  

  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.0"></script>



  



  




	





  





  

  
      <!-- UY BEGIN -->
      <script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid="></script>
      <!-- UY END -->
  




  
  

  

  

  

  


</body>
</html>
