<!doctype html>



  


<html class="theme-next muse use-motion" lang="en">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>



<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />












  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  
    
      
    

    
  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Arial, Lato:300,300italic,400,400italic,700,700italic|monospace, consolas, Courier New:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.0" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="CGmapTools, DNA methylation, Bisulfite sequencing, Allele specific methylation, SNP calling" />













<meta name="description" content="For users of CGmapTools
Any question or suggest or feedback is welcome!
Feel free to write to us! Your emails encourage us to improve the package.
We keep on updating selected and well-organized Q&amp;amp">
<meta property="og:type" content="website">
<meta property="og:title" content="Questions and Answers">
<meta property="og:url" content="https://cgmaptools.github.io/q-a/index.html">
<meta property="og:site_name" content="CGmapTools">
<meta property="og:description" content="For users of CGmapTools
Any question or suggest or feedback is welcome!
Feel free to write to us! Your emails encourage us to improve the package.
We keep on updating selected and well-organized Q&amp;amp">
<meta property="og:updated_time" content="2017-09-16T17:41:54.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Questions and Answers">
<meta name="twitter:description" content="For users of CGmapTools
Any question or suggest or feedback is welcome!
Feel free to write to us! Your emails encourage us to improve the package.
We keep on updating selected and well-organized Q&amp;amp">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Muse',
    sidebar: {"position":"left","display":"remove"},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: 'Author'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="https://cgmaptools.github.io/q-a/"/>





  <title>
  

  
    Questions and Answers | CGmapTools
  
</title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="en">

  










  
  
    
  

  <div class="container one-collumn sidebar-position-left  ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner">
<div class="site-meta custom-logo">
  
    <div class="site-meta-headline">
      <a>
        <img class="custom-logo-image" src="/images/CGmapTools_logo.png"
             alt="CGmapTools"/>
      </a>
    </div>
  

  <div class="custom-logo-site-title">
    <a href="/"  class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <span class="site-title">CGmapTools</span>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>
    
      <p class="site-subtitle">(40 functions for CGmap-oriented DNA methylation analyses)</p>
    
</div>

<div class="site-nav-toggle">
  <button>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
  </button>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            Home
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tutorial">
          <a href="/cgmaptools_documentation/what-is-cgmaptools.html" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-book"></i> <br />
            
            Tutorial
          </a>
        </li>
      
        
        <li class="menu-item menu-item-source code">
          <a href="/source-code" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-code"></i> <br />
            
            Source Code
          </a>
        </li>
      
        
        <li class="menu-item menu-item-release log">
          <a href="/release-log" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-list-ol"></i> <br />
            
            Release log
          </a>
        </li>
      
        
        <li class="menu-item menu-item-quick start">
          <a href="/quick-start" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-play"></i> <br />
            
            Quick Start
          </a>
        </li>
      
        
        <li class="menu-item menu-item-q&a">
          <a href="/q-a" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-question-circle"></i> <br />
            
            Q&A
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
  <header class="post-header">

	<h2 class="post-title" itemprop="name headline">Questions and Answers</h2>



</header>

    
    
      <h1 id="For-users-of-CGmapTools"><a href="#For-users-of-CGmapTools" class="headerlink" title="For users of CGmapTools"></a>For users of <em>CGmapTools</em></h1><ul>
<li>Any question or suggest or feedback is welcome!</li>
<li>Feel free to write to us! Your emails encourage us to improve the package.</li>
<li>We keep on updating selected and well-organized Q&amp;A on this page.</li>
<li>We welcome your evolvment/submission if you are interested to improve the code throught github.</li>
</ul>
<h1 id="Content"><a href="#Content" class="headerlink" title="Content"></a>Content</h1><ol>
<li>CGmap format<ul>
<li>1.1 Generate CGMapfiles from BAM files </li>
<li>1.2 CGmap format by BS-Seeker2 and CGmapTools</li>
</ul>
</li>
<li>Installation<ul>
<li>2.1 R program error “optparse”</li>
</ul>
</li>
<li>Differientially methylated region<ul>
<li>3.1 DMR analysis when serveral replicates</li>
<li>3.1 DMR analysis when lots of replicates</li>
<li>3.3 annotate DMRs</li>
</ul>
</li>
</ol>
<h1 id="Questions-and-Answers"><a href="#Questions-and-Answers" class="headerlink" title="Questions and Answers"></a>Questions and Answers</h1><h2 id="1-CGmap-format"><a href="#1-CGmap-format" class="headerlink" title="1. CGmap format"></a>1. CGmap format</h2><h3 id="1-1-Generate-CGmap-files-from-BAM-files"><a href="#1-1-Generate-CGmap-files-from-BAM-files" class="headerlink" title="1.1 Generate CGmap files from BAM files"></a>1.1 Generate CGmap files from BAM files</h3><p><strong>Question</strong> (by Zhao, Sep. 13<sup>th</sup>, 2017)</p>
<blockquote>
<p>Can I use <code>cgmaptools convert bam2cgmap</code> to generate CGmap files from the BAM files generated by <strong><em>BSMAP</em></strong>? </p>
</blockquote>
<p><strong>Answer</strong></p>
<blockquote>
<p><strong><em>BS-Seeker2</em></strong>, <strong><em>Bismark</em></strong> and <strong><em>BSMAP</em></strong> are the mainstream aligners for BS-seq data. We designed <strong><em>CGmapTools</em></strong> to be more general as a downstream analysis pipeline. </p>
<p>From our test, <strong><em>CGmapTools</em></strong> works in right-way for the BAM files generated by both <strong><em>BS-Seeker2</em></strong> and <strong><em>Bismark</em></strong>.  </p>
<p>However, it shows strange result for the BAM file <strong><em>BSMAP</em></strong> when generating ATCGmap files. It is not suggest to directly feed the BAM of <strong><em>BSMAP</em></strong> to <strong><em>CGmapTools</em></strong> currently.</p>
<p>Let us know if you have good solution.</p>
</blockquote>
<h3 id="1-2-CGmap-format-by-BS-Seeker2-and-CGmapTools"><a href="#1-2-CGmap-format-by-BS-Seeker2-and-CGmapTools" class="headerlink" title="1.2 CGmap format by BS-Seeker2 and CGmapTools"></a>1.2 CGmap format by <strong><em>BS-Seeker2</em></strong> and <strong><em>CGmapTools</em></strong></h3><p><strong>Question</strong> (by Fan, Mar. 5<sup>th</sup>, 2017)</p>
<blockquote>
<p>Here I know the <strong><em>BS-Seeker2</em></strong> and <strong><em>CGmapTools</em></strong> both can get the ATCGmap file and CGmap file. Are they the same file? </p>
</blockquote>
<p><strong>Answer</strong></p>
<blockquote>
<p>The file format is consistent between two softwares. </p>
<ul>
<li><strong><em>BS-Seeker2</em></strong> is implemented in python (depend on pysam package). </li>
<li><strong><em>CGmapTools</em></strong> implemented generating of CGmap from BAM in C language, which would be significantly faster for large genome. Also, <strong><em>CGmapTools</em></strong> supported the BAM files being generated by <strong><em>Bismark</em></strong>.</li>
</ul>
</blockquote>
<h2 id="2-Installation"><a href="#2-Installation" class="headerlink" title="2. Installation"></a>2. Installation</h2><h3 id="2-1-R-program-error-“optparse”"><a href="#2-1-R-program-error-“optparse”" class="headerlink" title="2.1 R program error “optparse”"></a>2.1 R program error “optparse”</h3><p><strong>Question</strong> (by Zhao, Sep. 13<sup>th</sup>, 2017)</p>
<blockquote>
<p>As the package <strong><em>optparse</em></strong> is required when running <code>cgmaptools lollipop</code>, I use <code>install.packages(&quot;optparse)</code> in R enviroment, but encounter the following error message<br>    <code>package \&#39;optparse\&#39; is not available (for R version x.x.x)</code></p>
</blockquote>
<p><strong>Answer</strong></p>
<blockquote>
<p>You can try to download the <strong><em>optparse</em></strong> to your local computer, and use <code>install.packages</code> to install the library locally.</p>
</blockquote>
<h2 id="3-Differientially-methylated-region"><a href="#3-Differientially-methylated-region" class="headerlink" title="3. Differientially methylated region"></a>3. Differientially methylated region</h2><h3 id="3-1-DMR-analysis-with-several-replicates"><a href="#3-1-DMR-analysis-with-several-replicates" class="headerlink" title="3.1 DMR analysis with several replicates"></a>3.1 DMR analysis with several replicates</h3><p><strong>Question</strong> (by Christelle, Aug. 1<sup>st</sup>, 2017)</p>
<blockquote>
<p>How could I used the DMR analysis function in <strong><em>CGmapTools</em></strong> if I have multiple replicates in both case and control? For example, 6 vs. 6.</p>
</blockquote>
<p><strong>Answer</strong></p>
<blockquote>
<p>The function <code>cgmaptools dmr</code> is designed to work with two <strong>CGmap</strong> files.<br>If you have multiple replicates, you can first merge the replicates to a single file (to increase coverage and detected region) use  the command <code>cgmaptools mergelist tosingle</code>.<br>Then use <code>cgmaptools dmr</code> for the two merged files.</p>
</blockquote>
<h3 id="3-1-DMR-analysis-with-lots-of-replicates"><a href="#3-1-DMR-analysis-with-lots-of-replicates" class="headerlink" title="3.1 DMR analysis with lots of replicates"></a>3.1 DMR analysis with lots of replicates</h3><p><strong>Question</strong> (by Yu-Ling, Sep. 5<sup>th</sup>, 2017)</p>
<blockquote>
<p>I worked on RRBS libraries, high-coverage for specific regions, 33 samples vs. 33 samples. How could I use <strong><em>CGmapTools</em></strong>?</p>
</blockquote>
<p><strong>Answer</strong></p>
<blockquote>
<p>When there are lots of replicates, and they are not sufferring from low-coverage, you can<br>(1) use the command <code>cgmaptools findCCGG</code> to find regions for RRBS liraries.<br>(2) use the command <code>cgmaptools mtr</code> to caculate the average mC levels in region region for each sample.<br>(3) merge result by (2) into a matrix<br>(4) write your own program to do <strong>t-test</strong>.</p>
</blockquote>
<h3 id="3-3-annotate-DMRs"><a href="#3-3-annotate-DMRs" class="headerlink" title="3.3 annotate DMRs"></a>3.3 annotate DMRs</h3><p><strong>Question</strong> (by Christelle, Sep. 16<sup>th</sup>, 2017)</p>
<blockquote>
<p>How could I annotate the DMR find by <strong><em>CGmapTools</em></strong>.</p>
</blockquote>
<p><strong>Answer</strong></p>
<blockquote>
<p>You may want to annotate DMRs to see the overlapping with genes, or specific functional regions. The software <strong><em>bedtools</em></strong> would be helpful for investigating overlaps.<br>If you find specific genes are intersting, you can visuazlise the DNA methylations levels, DMR or DMS in local region, using the command <code>cgmaptools lollipop</code>.</p>
</blockquote>

    
  </div>

          
          </div>
          


          
  <div class="comments" id="comments">
    
  </div>


        </div>
        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2017</span>
  <span class="with-love">
    <i class="fa fa-"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Weilong Guo and Ping Zhu</span>
</div>


<div class="powered-by">
  Powered by <a class="theme-link" href="https://hexo.io">Hexo</a>
</div>

<div class="theme-info">
  Theme -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Muse
  </a>
</div>


        

        
      </div>
    </footer>

    <div class="back-to-top">
      <i class="fa fa-arrow-up"></i>
    </div>
  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  




  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.0"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.0"></script>



  
  

  

  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.0"></script>



  



  




	





  





  

  
      <!-- UY BEGIN -->
      <script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid="></script>
      <!-- UY END -->
  




  
  

  

  

  

  


</body>
</html>
