<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  



<meta name="date" content="2018-08-27">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="what-is-cgmaptools.html">
<link rel="next" href="file-manipulation.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<h3 style="text-align:center;font-family:'Times New Roman'">CGmapTools Documentation</h3>
<a href="https://cgmaptools.github.io"><h4 style="text-align:center"><- Back Home</h4></a>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="what-is-cgmaptools.html"><a href="what-is-cgmaptools.html"><i class="fa fa-check"></i><b>1</b> What is CGmapTools</a></li>
<li class="chapter" data-level="2" data-path="file-formats.html"><a href="file-formats.html"><i class="fa fa-check"></i><b>2</b> File Formats</a><ul>
<li class="chapter" data-level="2.1" data-path="file-formats.html"><a href="file-formats.html#atcgmap-format"><i class="fa fa-check"></i><b>2.1</b> ATCGmap Format</a></li>
<li class="chapter" data-level="2.2" data-path="file-formats.html"><a href="file-formats.html#cgmap-format"><i class="fa fa-check"></i><b>2.2</b> CGmap Format</a></li>
<li class="chapter" data-level="2.3" data-path="file-formats.html"><a href="file-formats.html#atcgbz-format"><i class="fa fa-check"></i><b>2.3</b> ATCGbz Format</a></li>
<li class="chapter" data-level="2.4" data-path="file-formats.html"><a href="file-formats.html#cgbz-format"><i class="fa fa-check"></i><b>2.4</b> CGbz Format</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="file-manipulation.html"><a href="file-manipulation.html"><i class="fa fa-check"></i><b>3</b> File Manipulation</a><ul>
<li class="chapter" data-level="3.1" data-path="file-manipulation.html"><a href="file-manipulation.html#convert"><i class="fa fa-check"></i><b>3.1</b> convert</a></li>
<li class="chapter" data-level="3.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch"><i class="fa fa-check"></i><b>3.2</b> fetch</a><ul>
<li class="chapter" data-level="3.2.1" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch-cgbz"><i class="fa fa-check"></i><b>3.2.1</b> fetch cgbz</a></li>
<li class="chapter" data-level="3.2.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch-atcgbz"><i class="fa fa-check"></i><b>3.2.2</b> fetch atcgbz</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="file-manipulation.html"><a href="file-manipulation.html#refill"><i class="fa fa-check"></i><b>3.3</b> refill</a></li>
<li class="chapter" data-level="3.4" data-path="file-manipulation.html"><a href="file-manipulation.html#intersect"><i class="fa fa-check"></i><b>3.4</b> intersect</a></li>
<li class="chapter" data-level="3.5" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2"><i class="fa fa-check"></i><b>3.5</b> merge2</a><ul>
<li class="chapter" data-level="3.5.1" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2-atcgmap"><i class="fa fa-check"></i><b>3.5.1</b> merge2 atcgmap</a></li>
<li class="chapter" data-level="3.5.2" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2-cgmap"><i class="fa fa-check"></i><b>3.5.2</b> merge2 cgmap</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist"><i class="fa fa-check"></i><b>3.6</b> mergelist</a><ul>
<li class="chapter" data-level="3.6.1" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist-tomatrix"><i class="fa fa-check"></i><b>3.6.1</b> mergelist tomatrix</a></li>
<li class="chapter" data-level="3.6.2" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist-tosingle"><i class="fa fa-check"></i><b>3.6.2</b> mergelist tosingle</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="file-manipulation.html"><a href="file-manipulation.html#sort"><i class="fa fa-check"></i><b>3.7</b> sort</a></li>
<li class="chapter" data-level="3.8" data-path="file-manipulation.html"><a href="file-manipulation.html#split"><i class="fa fa-check"></i><b>3.8</b> split</a></li>
<li class="chapter" data-level="3.9" data-path="file-manipulation.html"><a href="file-manipulation.html#select"><i class="fa fa-check"></i><b>3.9</b> select</a><ul>
<li class="chapter" data-level="3.9.1" data-path="file-manipulation.html"><a href="file-manipulation.html#select-region"><i class="fa fa-check"></i><b>3.9.1</b> select region</a></li>
<li class="chapter" data-level="3.9.2" data-path="file-manipulation.html"><a href="file-manipulation.html#select-site"><i class="fa fa-check"></i><b>3.9.2</b> select site</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snv-calling.html"><a href="snv-calling.html"><i class="fa fa-check"></i><b>4</b> SNV calling</a><ul>
<li class="chapter" data-level="4.1" data-path="snv-calling.html"><a href="snv-calling.html#bayswc-strategy"><i class="fa fa-check"></i><b>4.1</b> BaysWC strategy</a></li>
<li class="chapter" data-level="4.2" data-path="snv-calling.html"><a href="snv-calling.html#binomwc-strategy"><i class="fa fa-check"></i><b>4.2</b> BinomWC strategy</a></li>
<li class="chapter" data-level="4.3" data-path="snv-calling.html"><a href="snv-calling.html#performance"><i class="fa fa-check"></i><b>4.3</b> Performance</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="methylation-analysis.html"><a href="methylation-analysis.html"><i class="fa fa-check"></i><b>5</b> Methylation Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dms"><i class="fa fa-check"></i><b>5.1</b> dms</a></li>
<li class="chapter" data-level="5.2" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dmr"><i class="fa fa-check"></i><b>5.2</b> dmr</a></li>
<li class="chapter" data-level="5.3" data-path="methylation-analysis.html"><a href="methylation-analysis.html#asm"><i class="fa fa-check"></i><b>5.3</b> asm</a></li>
<li class="chapter" data-level="5.4" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbed"><i class="fa fa-check"></i><b>5.4</b> mbed</a></li>
<li class="chapter" data-level="5.5" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbin"><i class="fa fa-check"></i><b>5.5</b> mbin</a></li>
<li class="chapter" data-level="5.6" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mmbin"><i class="fa fa-check"></i><b>5.6</b> mmbin</a></li>
<li class="chapter" data-level="5.7" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mfg"><i class="fa fa-check"></i><b>5.7</b> mfg</a></li>
<li class="chapter" data-level="5.8" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mstat"><i class="fa fa-check"></i><b>5.8</b> mstat</a></li>
<li class="chapter" data-level="5.9" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mtr"><i class="fa fa-check"></i><b>5.9</b> mtr</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="coverage-analysis.html"><a href="coverage-analysis.html"><i class="fa fa-check"></i><b>6</b> Coverage Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac"><i class="fa fa-check"></i><b>6.1</b> oac</a><ul>
<li class="chapter" data-level="6.1.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac-bin"><i class="fa fa-check"></i><b>6.1.1</b> oac bin</a></li>
<li class="chapter" data-level="6.1.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac-stat"><i class="fa fa-check"></i><b>6.1.2</b> oac stat</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec"><i class="fa fa-check"></i><b>6.2</b> mec</a><ul>
<li class="chapter" data-level="6.2.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec-bin"><i class="fa fa-check"></i><b>6.2.1</b> mec bin</a></li>
<li class="chapter" data-level="6.2.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec-stat"><i class="fa fa-check"></i><b>6.2.2</b> mec stat</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>7</b> Graphics</a><ul>
<li class="chapter" data-level="7.1" data-path="graphics.html"><a href="graphics.html#lollipop"><i class="fa fa-check"></i><b>7.1</b> lollipop</a></li>
<li class="chapter" data-level="7.2" data-path="graphics.html"><a href="graphics.html#heatmap"><i class="fa fa-check"></i><b>7.2</b> heatmap</a></li>
<li class="chapter" data-level="7.3" data-path="graphics.html"><a href="graphics.html#fragreg"><i class="fa fa-check"></i><b>7.3</b> fragreg</a></li>
<li class="chapter" data-level="7.4" data-path="graphics.html"><a href="graphics.html#tanghulu"><i class="fa fa-check"></i><b>7.4</b> tanghulu</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="other-ultilities.html"><a href="other-ultilities.html"><i class="fa fa-check"></i><b>8</b> Other Ultilities</a><ul>
<li class="chapter" data-level="8.1" data-path="other-ultilities.html"><a href="other-ultilities.html#findccgg"><i class="fa fa-check"></i><b>8.1</b> findCCGG</a></li>
<li class="chapter" data-level="8.2" data-path="other-ultilities.html"><a href="other-ultilities.html#bed2fragreg"><i class="fa fa-check"></i><b>8.2</b> bed2fragreg</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="file-formats" class="section level1">
<h1><span class="header-section-number">2</span> File Formats</h1>
<p>To facilitate high throughput data manipulation and reduce storage usage, several file format have been proposed and generaly accepted as the standard. Due to these great efforts (e.g.Â SAM/BAM and VCF), data analysis and tool development become more easier and highly efficient. However, when it comes to bisulfite sequencing data, currently, available tools possess their own tool specific data format. In consequence, integrating results from several tools leads to extra efforts in unifying data format and developing custermized tools, which is time comsuming and error prone.</p>
<p>The widely-used BS-seq alignment software <a href="http://pellegrini.mcdb.ucla.edu/BS_Seeker2/"><em><strong>BS-Seeker2</strong></em></a> defines <strong>CGmap</strong> and <strong>ATCGmap</strong> file formats for the representation of DNA methylomes. In CGmapTools, we used <strong>ATCGmap</strong> and <strong>CGmap</strong> as the standard file format interface, so that to simplify the development of downstream DNA methylation analysis tools and to provide standard formats for storing and sharing the DNA methylomes.</p>
<p>In <strong><em>CGmapTools</em></strong>, we designed novel binary formats: <strong>CGbz</strong> and <strong>ATCGbz</strong> for less coverage and improvements in random-accessing data in large data in hard-disk.</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-3"></span>
<img src="DataFormatSize.png" alt="Size of multiple file formats" width="60%" />
<p class="caption">
Figure 2.1: Size of multiple file formats
</p>
</div>
<div id="atcgmap-format" class="section level2">
<h2><span class="header-section-number">2.1</span> ATCGmap Format</h2>
<p>Similar with <a href="http://samtools.sourceforge.net/pileup.shtml"><strong>pileup</strong></a>, <strong>ATCGmap</strong> format summarizes the information of mapped reads covered on each nucleotide on both strands, specially designed for BS-seq data.</p>
<p>Here, we defined ATCGmap file format to integrate both mapping and coverage of non-cytosine and cytosine sites with estimated DNA methylation in a single file.</p>
<ul>
<li><p><strong>Example</strong></p>
<pre><code>chr1    T   3009410 --  --  0   10  0   0   0   0   3   0   0   0   na
chr1    C   3009411 CHH CC  0   10  0   0   0   0   4   0   0   0   0.0
chr1    C   3009412 CHG CC  0   10  0   0   0   0   9   1   0   0   0.0
chr1    C   3009413 CG  CG  0   10  50  0   0   0   20  1   0   0   0.83</code></pre></li>
<li><p><strong>Column Description</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-4"></span>
<img src="ATCGmap_Description.png" alt="Description of ATCGmap" width="90%" />
<p class="caption">
Figure 2.2: Description of ATCGmap
</p>
</div>
<!--```{r, echo=FALSE, results='asis'}
library(knitr)
atcgmap.format <- read.table("atcgmap.format.txt", header = T, sep = ";", quote = "")
names(atcgmap.format) <- c("Col", "Field", "Type", "Regexp/Range", "Brief description")
kable(atcgmap.format)
```-->
</div>
<div id="cgmap-format" class="section level2">
<h2><span class="header-section-number">2.2</span> CGmap Format</h2>
<p>In cases we only want to retain DNA methylation on cytonsines to save storage usage, we defined another file format called <strong>CGmap</strong> which provides sequence context and estimated DNA methylation level of any covered cytosines on the reference genome.</p>
<ul>
<li><p><strong>Example</strong></p>
<pre><code>chr1    G   3000851   CHH   CC  0.1   1   10
chr1    C   3001624   CHG   CA  0.0   0   9
chr1    C   3001631   CG    CG  1.0   5   5
chr1    G   3001632   CG    CG  0.9   9   10</code></pre></li>
<li><p><strong>Column Description</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-5"></span>
<img src="CGmap_Description.png" alt="Description of CGmap" width="90%" />
<p class="caption">
Figure 2.3: Description of CGmap
</p>
</div>
<!--```{r, echo=FALSE, results='asis'}
cgmap.format <- read.table("cgmap.format.txt", header = T, sep = ";", quote = "")
names(cgmap.format) <- c("Col", "Field", "Type", "Regexp/Range", "Brief description")
kable(cgmap.format)
```-->
</div>
<div id="atcgbz-format" class="section level2">
<h2><span class="header-section-number">2.3</span> ATCGbz Format</h2>
<p><strong>ATCGbz</strong> format is the binary compressed version for <strong>ATCGmap</strong> format. <strong>ATCGmap</strong> format is readable, while quite large for storing, and difficult for fetching information in a specific position. <strong>ATCGbz</strong> is defined as the sorted binary version, that storing all information of <strong>ATCGmap</strong> into standard binary form, largely reduced the storage requirement, and also supporting fast retrival of methylation information for any position on genome.</p>
<ul>
<li><strong>Data structure</strong></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-6"></span>
<img src="ATCGbz_structure1.png" alt="Data structure of ATCGbz" width="80%" />
<p class="caption">
Figure 2.4: Data structure of ATCGbz
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-7"></span>
<img src="ATCGbz_structure2.png" alt="Data structure of info field of ATCGbz" width="100%" />
<p class="caption">
Figure 2.5: Data structure of info field of ATCGbz
</p>
</div>
<ul>
<li>Related command</li>
</ul>
<p><strong>Command</strong></p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> fetch atcgbz -h </code></pre></div>
<pre><code>#   
#     Usage: cgmaptools fetch atcgbz -b &lt;ATCGbz&gt; -C &lt;CHR&gt; -L &lt;LeftPos&gt; -R &lt;RightPos&gt;
#           (aka ATCGbzFetchRegion)
#     Description: Convert ATCGbz format to ATCGmap format.
#     Contact:     Guo, Weilong; guoweilong@126.com
#     Last update: 2016-12-07
#   
#     Options:
#   
#       -h, --help             output help information
#       -b, --ATCGbz &lt;arg&gt;     output ATCGbz file
#       -C, --CHR &lt;arg&gt;        specify the chromosome name
#       -L, --leftPos &lt;arg&gt;    the left position
#       -R, --rightPos &lt;arg&gt;   the right position</code></pre>
</div>
<div id="cgbz-format" class="section level2">
<h2><span class="header-section-number">2.4</span> CGbz Format</h2>
<p><strong>CGbz</strong> format is the binary compressed version for <strong>CGmap</strong> format.</p>
<ul>
<li><strong>Data structure</strong></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-9"></span>
<img src="CGbz_structure1.png" alt="Data structure of ATCGbz" width="80%" />
<p class="caption">
Figure 2.6: Data structure of ATCGbz
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-10"></span>
<img src="CGbz_structure2.png" alt="Data structure of info field of CGbz" width="60%" />
<p class="caption">
Figure 2.7: Data structure of info field of CGbz
</p>
</div>
<ul>
<li>Related command</li>
</ul>
<p><strong>Command</strong></p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> fetch cgbz -h </code></pre></div>
<pre><code>#   
#     Usage: cgmaptools fetch cgbz -b &lt;CGbz&gt; -C &lt;CHR&gt; -L &lt;LeftPos&gt; -R &lt;RightPos&gt;
#            (aka CGvzFetchRegion)
#     Description: Convert CGbz file to CGmap format.
#     Contact: Guo, Weilong; guoweilong@126.com
#     Last update: 2016-12-07
#   
#     Options:
#   
#       -h, --help             output help information
#       -b, --CGbz &lt;arg&gt;       output CGbz file
#       -C, --CHR &lt;arg&gt;        specify the chromosome name
#       -L, --leftPos &lt;arg&gt;    the left position
#       -R, --rightPos &lt;arg&gt;   the right position</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="what-is-cgmaptools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="file-manipulation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"download": ["CGmapTools_documentation.pdf", "CGmapTools_documentation.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
