<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations.">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  



<meta name="date" content="2017-02-20">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="new-file-format.html">
<link rel="next" href="snp-calling.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<h3 style="text-align:center;font-family:'Times New Roman'">CGmapTools Documentation</h3>
<a href="https://cgmaptools.github.io"><h4 style="text-align:center"><- Back Home</h4></a>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="what-is-cgmaptools.html"><a href="what-is-cgmaptools.html"><i class="fa fa-check"></i><b>1</b> What is CGmapTools</a></li>
<li class="chapter" data-level="2" data-path="new-file-format.html"><a href="new-file-format.html"><i class="fa fa-check"></i><b>2</b> New File Format</a><ul>
<li class="chapter" data-level="2.1" data-path="new-file-format.html"><a href="new-file-format.html#atcgmap-format"><i class="fa fa-check"></i><b>2.1</b> ATCGmap Format</a></li>
<li class="chapter" data-level="2.2" data-path="new-file-format.html"><a href="new-file-format.html#cgmap-format"><i class="fa fa-check"></i><b>2.2</b> CGmap Format</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="file-manipulation.html"><a href="file-manipulation.html"><i class="fa fa-check"></i><b>3</b> File Manipulation</a><ul>
<li class="chapter" data-level="3.1" data-path="file-manipulation.html"><a href="file-manipulation.html#convert"><i class="fa fa-check"></i><b>3.1</b> convert</a></li>
<li class="chapter" data-level="3.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch"><i class="fa fa-check"></i><b>3.2</b> fetch</a></li>
<li class="chapter" data-level="3.3" data-path="file-manipulation.html"><a href="file-manipulation.html#refill"><i class="fa fa-check"></i><b>3.3</b> refill</a></li>
<li class="chapter" data-level="3.4" data-path="file-manipulation.html"><a href="file-manipulation.html#intersect"><i class="fa fa-check"></i><b>3.4</b> intersect</a></li>
<li class="chapter" data-level="3.5" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2"><i class="fa fa-check"></i><b>3.5</b> merge2</a></li>
<li class="chapter" data-level="3.6" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist"><i class="fa fa-check"></i><b>3.6</b> mergelist</a></li>
<li class="chapter" data-level="3.7" data-path="file-manipulation.html"><a href="file-manipulation.html#sort"><i class="fa fa-check"></i><b>3.7</b> sort</a></li>
<li class="chapter" data-level="3.8" data-path="file-manipulation.html"><a href="file-manipulation.html#split"><i class="fa fa-check"></i><b>3.8</b> split</a></li>
<li class="chapter" data-level="3.9" data-path="file-manipulation.html"><a href="file-manipulation.html#select"><i class="fa fa-check"></i><b>3.9</b> select</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snp-calling.html"><a href="snp-calling.html"><i class="fa fa-check"></i><b>4</b> SNP calling</a></li>
<li class="chapter" data-level="5" data-path="methylation-analysis.html"><a href="methylation-analysis.html"><i class="fa fa-check"></i><b>5</b> Methylation Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dms"><i class="fa fa-check"></i><b>5.1</b> dms</a></li>
<li class="chapter" data-level="5.2" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dmr"><i class="fa fa-check"></i><b>5.2</b> dmr</a></li>
<li class="chapter" data-level="5.3" data-path="methylation-analysis.html"><a href="methylation-analysis.html#asm"><i class="fa fa-check"></i><b>5.3</b> asm</a></li>
<li class="chapter" data-level="5.4" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbed"><i class="fa fa-check"></i><b>5.4</b> mbed</a></li>
<li class="chapter" data-level="5.5" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbin"><i class="fa fa-check"></i><b>5.5</b> mbin</a></li>
<li class="chapter" data-level="5.6" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mmbin"><i class="fa fa-check"></i><b>5.6</b> mmbin</a></li>
<li class="chapter" data-level="5.7" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mfg"><i class="fa fa-check"></i><b>5.7</b> mfg</a></li>
<li class="chapter" data-level="5.8" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mstat"><i class="fa fa-check"></i><b>5.8</b> mstat</a></li>
<li class="chapter" data-level="5.9" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mtr"><i class="fa fa-check"></i><b>5.9</b> mtr</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="coverage-analysis.html"><a href="coverage-analysis.html"><i class="fa fa-check"></i><b>6</b> Coverage Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac"><i class="fa fa-check"></i><b>6.1</b> oac</a></li>
<li class="chapter" data-level="6.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec"><i class="fa fa-check"></i><b>6.2</b> mec</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>7</b> Graphics</a><ul>
<li class="chapter" data-level="7.1" data-path="graphics.html"><a href="graphics.html#lollipop"><i class="fa fa-check"></i><b>7.1</b> lollipop</a></li>
<li class="chapter" data-level="7.2" data-path="graphics.html"><a href="graphics.html#heatmap"><i class="fa fa-check"></i><b>7.2</b> heatmap</a></li>
<li class="chapter" data-level="7.3" data-path="graphics.html"><a href="graphics.html#fragreg"><i class="fa fa-check"></i><b>7.3</b> fragreg</a></li>
<li class="chapter" data-level="7.4" data-path="graphics.html"><a href="graphics.html#tanghulu"><i class="fa fa-check"></i><b>7.4</b> tanghulu</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="other-ultilities.html"><a href="other-ultilities.html"><i class="fa fa-check"></i><b>8</b> Other Ultilities</a><ul>
<li class="chapter" data-level="8.1" data-path="other-ultilities.html"><a href="other-ultilities.html#findccgg"><i class="fa fa-check"></i><b>8.1</b> findCCGG</a></li>
<li class="chapter" data-level="8.2" data-path="other-ultilities.html"><a href="other-ultilities.html#bed2fragreg"><i class="fa fa-check"></i><b>8.2</b> bed2fragreg</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="file-manipulation" class="section level1">
<h1><span class="header-section-number">3</span> File Manipulation</h1>
<p><strong>CGmapTools</strong> provides multiple utilities to manipulate files in ATCGmap and CGmap format or compressed ATCGbz/CGbz format.</p>
<p><strong>Usage</strong>: <code>cgmaptools &lt;convert|fetch|refill|intersect|merge2|mergelist|sort|split|select|&gt; [options]</code></p>
<div id="convert" class="section level2">
<h2><span class="header-section-number">3.1</span> convert</h2>
<ul>
<li><p><strong>Description</strong>: File format coversion.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools convert &lt;command&gt; [options]</code></p></li>
<li><p><strong>Commands</strong>:</p></li>
</ul>
<table>
<thead>
<tr class="header">
<th align="center">Commands</th>
<th align="center">From</th>
<th align="center">To</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">bam2cgmap</td>
<td align="center">BAM</td>
<td align="center">CGmap &amp; ATCGmap</td>
</tr>
<tr class="even">
<td align="center">atcgmap2atcgbz</td>
<td align="center">ATCGmap</td>
<td align="center">ATCGbz</td>
</tr>
<tr class="odd">
<td align="center">atcgbz2atcgmap</td>
<td align="center">ATCGbz</td>
<td align="center">ATCGmap</td>
</tr>
<tr class="even">
<td align="center">atcgmap2cgmap</td>
<td align="center">ATCGmap</td>
<td align="center">CGmap</td>
</tr>
<tr class="odd">
<td align="center">cgmap2cgbz</td>
<td align="center">CGamp</td>
<td align="center">CGbz</td>
</tr>
<tr class="even">
<td align="center">cgbz2cgmap</td>
<td align="center">CGbz</td>
<td align="center">CGmap</td>
</tr>
<tr class="odd">
<td align="center">cgmap2wig</td>
<td align="center">CGmap</td>
<td align="center">WIG</td>
</tr>
</tbody>
</table>
<ul>
<li><p><strong>Example</strong>:</p>
<p>#1 The commands below will covert bam file to cgmap format.</p>
<p><code>cgmaptools convert bam2cgmap -b &lt;BAM&gt; -g &lt;genome.fa&gt; -o &lt;prefix&gt;</code></p>
<p>#2 This command will convert cgmap to wig format.</p>
<p><code>cgmaptools convert cgmap2wig [-i &lt;CGmap&gt;] [-w &lt;wig&gt;] [-c &lt;INT&gt; -b &lt;float&gt;]</code></p></li>
</ul>
<blockquote>
<p>Note: please refer to the help message for usage details using <code>-h</code> option.</p>
</blockquote>
</div>
<div id="fetch" class="section level2">
<h2><span class="header-section-number">3.2</span> fetch</h2>
<ul>
<li><p><strong>Description</strong>: Fastly acess methylation data in specified region.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools fetch &lt;command&gt; [options]</code></p></li>
<li><p><strong>Commands</strong>:</p>
<p><strong>atcgbz</strong>: fetch lines from ATCGbz file.</p>
<p>Usage: <code>cgmaptools fetch atcgbz -b &lt;ATCGbz&gt; -C &lt;CHR&gt; -L &lt;LeftPos&gt; -R &lt;RightPos&gt;</code></p>
<pre><code>-b, --ATCGbz &lt;arg&gt;     input ATCGbz file  
-C, --CHR &lt;arg&gt;        specify the chromosome name  
-L, --leftPos &lt;arg&gt;    the left position  
-R, --rightPos &lt;arg&gt;   the right position</code></pre>
<p><strong>cgbz</strong>: fetch lines from CGbz file.</p>
<p>Usage: <code>cgmaptools fetch cgbz -b &lt;CGbz&gt; -C &lt;CHR&gt; -L &lt;LeftPos&gt; -R &lt;RightPos&gt;</code></p>
<pre><code>-b, --CGbz &lt;arg&gt;       intput CGbz file
-C, --CHR &lt;arg&gt;        specify the chromosome name
-L, --leftPos &lt;arg&gt;    the left position
-R, --rightPos &lt;arg&gt;   the right position</code></pre></li>
</ul>
</div>
<div id="refill" class="section level2">
<h2><span class="header-section-number">3.3</span> refill</h2>
<ul>
<li><p><strong>Description</strong>: Fill the CG/CHG/CHH and CA/CC/CT/CG context to CGmap or ATCGmap files. Other fields will not be affected.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools refill [-i &lt;CGmap&gt;] -g &lt;genome.fa&gt; [-o output]</code></p>
<pre><code>-i STRING     Input CGmap file (CGmap or CGmap.gz)
-g STRING     genome file, FASTA format (gzipped if end with &#39;.gz&#39;)
-o STRING     Output file name (gzipped if end with &#39;.gz&#39;)
-0, --0-base  0-based genome if specified [Default: 1-based]</code></pre></li>
<li><p><strong>Example</strong>:</p>
<p>The input CGmap file, which is lacking C context on the 3rd and 4th columns:</p>
<pre><code>Chr1    C       3541    -       -       0.0     0       1</code></pre>
<p>After <code>refill</code> processing, the CGmap file would be as below, added C context information:</p>
<pre><code>Chr1    C       3541    CG      CG      0.0     0       1</code></pre></li>
</ul>
</div>
<div id="intersect" class="section level2">
<h2><span class="header-section-number">3.4</span> intersect</h2>
<ul>
<li><p><strong>Description</strong>: Get the intersection of two CGmap files.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools intersect [-1 &lt;CGmap_1&gt;] -2 &lt;CGmap_2&gt; [-o &lt;output&gt;]</code></p>
<pre><code>-1 CGmap File  File name, end with .CGmap or .CGmap.gz.
-2 CGmap File  standard input if not specified
-o OUTFILE     To standard output if not specified. Compressed output if end
               with .gz</code></pre></li>
<li><p><strong>Example</strong>:</p>
<p>Suppose you have two CGmap file from two samples, the first one is:</p>
<pre><code>Chr1  C  3541  CG  CG  0.8  4  5</code></pre>
<p>and the second CGmap file is:</p>
<pre><code>Chr1  C  3541  CG  CG  0.4  4  10</code></pre>
<p>After intersction, the output contains sites covered in both CGmap files. And the last three columns of the output are extracted from the second CGmap file:</p>
<pre><code>Chr1  C  3541  CG  CG  0.8  4  5  0.4  4  10</code></pre></li>
</ul>
</div>
<div id="merge2" class="section level2">
<h2><span class="header-section-number">3.5</span> merge2</h2>
<ul>
<li><p><strong>Description</strong>: Merge two CGmap or ATCGmap files together.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools merge2 &lt;command&gt; [options]</code></p></li>
<li><p><strong>Commands</strong>:</p>
<p><strong>atcgmap</strong>: merge two ATCGmap files into one.</p>
<p>Usage: <code>cgmaptools merge2 atcgmap -1 &lt;ATCGmap&gt; -2 &lt;ATCGmap&gt;</code></p>
<pre><code>-1  Input, 1st ATCGmap file
-2  Input, 2nd ATCGmap file
Output to STDOUT in ATCGmap format
Tips: Two input files should have the same order of chromosomes</code></pre>
<p><strong>cgmap</strong>: merge two CGmap files into one.</p>
<p>Usage: <code>cgmaptools merge2 cgmap -1 &lt;CGmap_1&gt; -2 &lt;CGmap_2&gt; [-o &lt;output&gt;]</code></p>
<pre><code>-1 FILE     File name end with .CGmap or .CGmap.gz
-2 FILE     If not specified, STDIN will be used.
-o OUTFILE  CGmap, output file. Use STDOUT if omitted (gzipped if end with
            &#39;.gz&#39;).</code></pre></li>
</ul>
</div>
<div id="mergelist" class="section level2">
<h2><span class="header-section-number">3.6</span> mergelist</h2>
<ul>
<li><p><strong>Description</strong>: merge a list of files.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools mergelist &lt;command&gt; [options]</code></p></li>
<li><p><strong>Commands</strong>:</p>
<p><strong>tomatrix</strong>: Fill methylation levels according to the Index file for CGmap files in list.</p>
<p>Usage: <code>cgmaptools mergelist tomatrix  [-i &lt;index&gt;] -f &lt;IN1,IN2,..&gt; -t &lt;tag1,tag2,..&gt; [-o output]</code></p>
<pre><code>-i FILE     TXT file, index file, use STDIN if omitted
-f STRING   List of (input) CGmap files (CGmap or CGmap.gz)
-t STRING   List of tags, same order with &#39;-f&#39;
-c INT      minimum coverage [default: 1]
-C INT      maximum coverage [default: 200]
-o STRING   Output file name (gzipped if end with &#39;.gz&#39;)</code></pre>
<p><strong>tosingle</strong>: merge list of input files into one.</p>
<p>Usage: <code>cgmaptools mergelist tosingle -i f1,f2,..,fn [-o &lt;output&gt;]</code></p>
<pre><code>-i FILE     List of input files; gzipped file ends with &#39;.gz&#39;
-f FILE     cgmap or atcgmap [Default: cgmap]
-o OUTFILE  To standard output if not specified; gzipped file if end with
            &#39;.gz&#39;</code></pre></li>
</ul>
</div>
<div id="sort" class="section level2">
<h2><span class="header-section-number">3.7</span> sort</h2>
<ul>
<li><p><strong>Description</strong>: Sort the input files by chromosome and position.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools sort [-i &lt;input&gt;] [-c 1] [-p 3] [-o output]</code></p>
<pre><code>-i FILE            File name end with .CGmap or .CGmap.gz. 
                   If not specified, STDIN will be used.
-c INT, --chr=INT  The column of chromosome [default: 1]
-p INT, --pos=INT  The column of position [default: 2]
-o OUTFILE         To standard output if not specified</code></pre></li>
</ul>
</div>
<div id="split" class="section level2">
<h2><span class="header-section-number">3.8</span> split</h2>
<ul>
<li><p><strong>Description</strong>: Split the files by each chromosomes.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools split -i &lt;input&gt; -p &lt;prefix[.chr.]&gt; -s &lt;[.chr.]suffix&gt;</code></p>
<pre><code>-i FILE     Input file, CGmap or ATCGmap foramt, use STDIN when not specified.
            (gzipped if end with &#39;gz&#39;).
-p STRING   The prefix for output file
-s STRING   The suffix for output file (gzipped if end with &#39;gz&#39;).</code></pre></li>
</ul>
</div>
<div id="select" class="section level2">
<h2><span class="header-section-number">3.9</span> select</h2>
<ul>
<li><p><strong>Description</strong>: Split the files by each chromosomes.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools select &lt;command&gt; [options]</code></p></li>
<li><p><strong>Commands</strong>:</p>
<p><strong>region</strong>: Lines in input CGmap/ATCGmap be selected/excluded by BED file. Strand is NOT considered. Output to STDOUT in same format with input.</p>
<p>Usage: <code>cgmaptools select region [-i &lt;CGmap/ATCGmap&gt;] -r &lt;BED&gt; [-R]</code></p>
<pre><code>-i  Input, CGmap/ATCGmap file; use STDIN if not specified
    Please use &quot;gunzip -c &lt;input&gt;.gz &quot; and pipe as input for gzipped file.
    Ex: chr12   G   19898796    ...
-r  Input, Region file, BED file to store regions
    At least 3 columns are required
    Ex: chr12 19898766 19898966 XX XXX XXX
-R  [optional] Reverse selection. Sites in region file will be excluded when specified</code></pre>
<p><strong>site</strong>: Select lines from input CGmap/ATCGmap in index or reverse.</p>
<p>Usage: <code>cgmaptools select site -i &lt;index&gt; [-f &lt;CGmap/ATCGmap&gt;] [-r] [-o output]</code></p>
<pre><code>-i FILE     Name of Index file required (gzipped if end with &#39;.gz&#39;).
-r          reverse selected, remove site in index if specified
-f STRING   Input CGmap/ATCGmap files. Use STDIN if not specified
-o STRING   CGmap, Output file name (gzipped if end with &#39;.gz&#39;).</code></pre></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="new-file-format.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="snp-calling.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"download": ["CGmapTools_documentation.pdf", "CGmapTools_documentation.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

</body>

</html>
