<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations.">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  



<meta name="date" content="2017-02-20">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="coverage-analysis.html">
<link rel="next" href="other-ultilities.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<h3 style="text-align:center;font-family:'Times New Roman'">CGmapTools Documentation</h3>
<a href="https://cgmaptools.github.io"><h4 style="text-align:center"><- Back Home</h4></a>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="what-is-cgmaptools.html"><a href="what-is-cgmaptools.html"><i class="fa fa-check"></i><b>1</b> What is CGmapTools</a></li>
<li class="chapter" data-level="2" data-path="new-file-format.html"><a href="new-file-format.html"><i class="fa fa-check"></i><b>2</b> New File Format</a><ul>
<li class="chapter" data-level="2.1" data-path="new-file-format.html"><a href="new-file-format.html#atcgmap-format"><i class="fa fa-check"></i><b>2.1</b> ATCGmap Format</a></li>
<li class="chapter" data-level="2.2" data-path="new-file-format.html"><a href="new-file-format.html#cgmap-format"><i class="fa fa-check"></i><b>2.2</b> CGmap Format</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="file-manipulation.html"><a href="file-manipulation.html"><i class="fa fa-check"></i><b>3</b> File Manipulation</a><ul>
<li class="chapter" data-level="3.1" data-path="file-manipulation.html"><a href="file-manipulation.html#convert"><i class="fa fa-check"></i><b>3.1</b> convert</a></li>
<li class="chapter" data-level="3.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch"><i class="fa fa-check"></i><b>3.2</b> fetch</a></li>
<li class="chapter" data-level="3.3" data-path="file-manipulation.html"><a href="file-manipulation.html#refill"><i class="fa fa-check"></i><b>3.3</b> refill</a></li>
<li class="chapter" data-level="3.4" data-path="file-manipulation.html"><a href="file-manipulation.html#intersect"><i class="fa fa-check"></i><b>3.4</b> intersect</a></li>
<li class="chapter" data-level="3.5" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2"><i class="fa fa-check"></i><b>3.5</b> merge2</a></li>
<li class="chapter" data-level="3.6" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist"><i class="fa fa-check"></i><b>3.6</b> mergelist</a></li>
<li class="chapter" data-level="3.7" data-path="file-manipulation.html"><a href="file-manipulation.html#sort"><i class="fa fa-check"></i><b>3.7</b> sort</a></li>
<li class="chapter" data-level="3.8" data-path="file-manipulation.html"><a href="file-manipulation.html#split"><i class="fa fa-check"></i><b>3.8</b> split</a></li>
<li class="chapter" data-level="3.9" data-path="file-manipulation.html"><a href="file-manipulation.html#select"><i class="fa fa-check"></i><b>3.9</b> select</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snv-calling.html"><a href="snv-calling.html"><i class="fa fa-check"></i><b>4</b> SNV calling</a></li>
<li class="chapter" data-level="5" data-path="methylation-analysis.html"><a href="methylation-analysis.html"><i class="fa fa-check"></i><b>5</b> Methylation Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dms"><i class="fa fa-check"></i><b>5.1</b> dms</a></li>
<li class="chapter" data-level="5.2" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dmr"><i class="fa fa-check"></i><b>5.2</b> dmr</a></li>
<li class="chapter" data-level="5.3" data-path="methylation-analysis.html"><a href="methylation-analysis.html#asm"><i class="fa fa-check"></i><b>5.3</b> asm</a></li>
<li class="chapter" data-level="5.4" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbed"><i class="fa fa-check"></i><b>5.4</b> mbed</a></li>
<li class="chapter" data-level="5.5" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbin"><i class="fa fa-check"></i><b>5.5</b> mbin</a></li>
<li class="chapter" data-level="5.6" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mmbin"><i class="fa fa-check"></i><b>5.6</b> mmbin</a></li>
<li class="chapter" data-level="5.7" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mfg"><i class="fa fa-check"></i><b>5.7</b> mfg</a></li>
<li class="chapter" data-level="5.8" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mstat"><i class="fa fa-check"></i><b>5.8</b> mstat</a></li>
<li class="chapter" data-level="5.9" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mtr"><i class="fa fa-check"></i><b>5.9</b> mtr</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="coverage-analysis.html"><a href="coverage-analysis.html"><i class="fa fa-check"></i><b>6</b> Coverage Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac"><i class="fa fa-check"></i><b>6.1</b> oac</a></li>
<li class="chapter" data-level="6.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec"><i class="fa fa-check"></i><b>6.2</b> mec</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>7</b> Graphics</a><ul>
<li class="chapter" data-level="7.1" data-path="graphics.html"><a href="graphics.html#lollipop"><i class="fa fa-check"></i><b>7.1</b> lollipop</a></li>
<li class="chapter" data-level="7.2" data-path="graphics.html"><a href="graphics.html#heatmap"><i class="fa fa-check"></i><b>7.2</b> heatmap</a></li>
<li class="chapter" data-level="7.3" data-path="graphics.html"><a href="graphics.html#fragreg"><i class="fa fa-check"></i><b>7.3</b> fragreg</a></li>
<li class="chapter" data-level="7.4" data-path="graphics.html"><a href="graphics.html#tanghulu"><i class="fa fa-check"></i><b>7.4</b> tanghulu</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="other-ultilities.html"><a href="other-ultilities.html"><i class="fa fa-check"></i><b>8</b> Other Ultilities</a><ul>
<li class="chapter" data-level="8.1" data-path="other-ultilities.html"><a href="other-ultilities.html#findccgg"><i class="fa fa-check"></i><b>8.1</b> findCCGG</a></li>
<li class="chapter" data-level="8.2" data-path="other-ultilities.html"><a href="other-ultilities.html#bed2fragreg"><i class="fa fa-check"></i><b>8.2</b> bed2fragreg</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="graphics" class="section level1">
<h1><span class="header-section-number">7</span> Graphics</h1>
<div id="lollipop" class="section level2">
<h2><span class="header-section-number">7.1</span> lollipop</h2>
<ul>
<li><p><strong>Description</strong>: Plot local mC level for multiple samples.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools lollipop [options] file</code></p>
<pre><code>-i INFILE, --infile=INFILE
      input file
-a ANNOTATION, --annotation=ANNOTATION
    [opt] sample name
-o OUTFILE, --outfile=OUTFILE
    [opt] output file
-f FORMAT, --format=FORMAT
    [opt] the format for output figure: pdf (default), png, eps
-l LEFT, --left=LEFT
    [opt] Left-most position
-r RIGHT, --right=RIGHT
    [opt] Right-most position
-c CHR, --chr=CHR
    [opt] chromosome name
-t TITLE, --title=TITLE
    [opt] text shown on title
-w WIDTH, --width=WIDTH
    [opt] width (in inch). Default: 8.
--height=HEIGHT
    [opt] height (in inch). Default: 8.
-s SITE, --site=SITE
    [opt] file of site to be marked
-b BED, --bed=BED
    [opt] BED file for region to be markered</code></pre></li>
<li><p><strong>Example command</strong>:</p>
<p><code>cgmaptools lollipop -i matrix.CG.gz -a anno.refFlat -f pdf</code></p></li>
<li><p><strong>File format</strong>:</p>
<p>The input file format:</p>
<blockquote>
<p>&gt;= 3 columns, 1st line is the header, using R color name or “NaN”. Can be output by CGmapFillIndex.py. Use STDIN if omitted.</p>
</blockquote>
<pre><code>chr   pos       E_vs_EMT    EMT_vs_M    E_vs_M
chr1    13116801    NaN       NaN         darkgreen
chr1    13116899    NaN       red         NaN</code></pre>
<p>The bed file format:</p>
<blockquote>
<p>the first 4 columns are required.</p>
</blockquote>
<pre><code>chr1  213941196  213942363  REGION-1
chr1  213942363  213943530  REGION-2</code></pre></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-10"></span>
<img src="TRIM59.png" alt="Lollipop example-1" width="80%" />
<p class="caption">
Figure 7.1: Lollipop example-1
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-11"></span>
<img src="VCAN.png" alt="Lollipop example-2" width="80%" />
<p class="caption">
Figure 7.2: Lollipop example-2
</p>
</div>
</div>
<div id="heatmap" class="section level2">
<h2><span class="header-section-number">7.2</span> heatmap</h2>
<ul>
<li><p><strong>Description</strong>: Plot methylation dynamics of target region for multiple samples <a href="graphics.html#heatmap">heatmap</a>.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools heatmap [options]</code></p>
<pre><code>-i INFILE, --infile=INFILE
    input file
-o OUTFILE, --outfile=OUTFILE
    [opt] output file name. [default: mCBinHeatmap.SysDate.pdf]
-c, --cluster
    [opt] cluster samples by methylation in regions. [default: FALSE]
-l COLORLOW, --colorLow=COLORLOW
    [opt] color used for the lowest methylation value. [default: cyan3]
-m COLORMID, --colorMid=COLORMID
    [opt] color used for the middle methylation value. [default: null]
-b COLORHIGH, --colorHigh=COLORHIGH
    [opt] color used for the highest methylation value. [default: coral2]
-n COLORNUMBER, --colorNumber=COLORNUMBER
    [opt] desired number of color elements in the panel. [default: 10]
-W WIDTH, --width=WIDTH
    [opt] width of figure (inch). [default: 7]
-H HEIGHT, --height=HEIGHT
    [opt] height of figure (inch). [default: 7]
-f FORMAT, --format=FORMAT
    [opt] format of output figure. Alternative: png. [default: pdf]
-R RESOLUTION, --resolution=RESOLUTION
    [opt] Resolution in ppi. Only available for png format. [default: 300]</code></pre></li>
<li><p><strong>Example command</strong>:</p>
<p><code>cgmaptools heatmap -i mmbin -c -o cluster.pdf -f pdf</code></p></li>
<li><p><strong>File format</strong>:</p>
<p>The input file format:</p>
<blockquote>
<p>The 1st line is the header. Each column contains methylation measurements of a sample.</p>
</blockquote>
<pre><code>Region  Sample1  Sample2 ...
Region1 0.1      0.1     ...
Region2 0.1      0.1     ...</code></pre></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-12"></span>
<img src="mC_Cluster.png" alt="heatmap example-1" width="80%" />
<p class="caption">
Figure 7.3: heatmap example-1
</p>
</div>
</div>
<div id="fragreg" class="section level2">
<h2><span class="header-section-number">7.3</span> fragreg</h2>
<ul>
<li><p><strong>Description</strong>: Plot methylation dynamics of target and flanking region for multiple samples.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools fragreg [options]</code></p>
<pre><code>-i INFILE, --infile=INFILE
    input file
-r RATIO, --ratio=RATIO
    [opt] range ratio between target region and flanking region in plot. [default: 5]
-o OUTFILE, --outfile=OUTFILE
    [opt] output file name. [default: FragRegView.SysDate.pdf]
-W WIDTH, --width=WIDTH
    [opt] width of figure (inch). [default: 7]
-H HEIGHT, --height=HEIGHT
    [opt] height of figure (inch). [default: 7]
-f FORMAT, --format=FORMAT
    [opt] format of output figure. Alternative: png. [default: pdf]
-R RESOLUTION, --resolution=RESOLUTION
    [opt] Resolution in ppi. Only available for png format. [default: 300]</code></pre></li>
<li><p><strong>File format</strong>:</p>
<p>The input file format:</p>
<blockquote>
<p>The 1st line is the header. Each row contains methylation measurements of a sample.</p>
</blockquote>
<pre><code>Sample  Up1  Up2  ...  Region1  Region2 ...  Down1  Down2  ...
Sample1 0.1  0.1  ...  0.2      0.2     ...  0.3    0.3    ...
Sample2 0.1  0.1  ...  0.2      0.2     ...  0.3    0.3    ...</code></pre></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-13"></span>
<img src="GeneBody_mC_profile.png" alt="FragRegMC example" width="80%" />
<p class="caption">
Figure 7.4: FragRegMC example
</p>
</div>
</div>
<div id="tanghulu" class="section level2">
<h2><span class="header-section-number">7.4</span> tanghulu</h2>
<ul>
<li><p><strong>Description</strong>: Show local mapped reads in Tanghulu shape.</p></li>
<li><p><strong>Usage</strong>: <code>cgmaptools tanghulu [options] -r &lt;ref&gt; -b &lt;bam&gt; -l chr1:133-144</code></p>
<pre><code>-r    Samtools indexed reference genome seqeunce, fasta format. eg. hg19.fa
      - use samtools to index reference: samtools faidx &lt;hg19.fa&gt;
-b    Samtools indexed Bam file to view.
      - use samtools to index bam file: samtools index &lt;input.bam&gt;
-l    Region in which to display DNA methylation.
      - or specify a single position (eg. heterozygous SNP site), 
        we will show allele specific methylation.
-s    Path to samtools eg. /home/user/bin/samtools
      - by defualt, we try to search samtools in your system PATH.
-o    Output results to file [default: CirclePlot.Ctype.region.Date.pdf].
-t    C context. [default: CG]
      - available context: C, CG, CH, CW, CC, CA, CT, CHG, CHH
-d    Ouput device. [default: pdf]
      - alternative: png
-c    Seperate reads by chain. [default: OFF]
      - specify this option to turn ON.
-v    Show vague allele linked reads. [ default: OFF]
-g    Genotype of heterozygous SNP site.
      - This option provides two alleles of htSNP site. eg. AT
      - The genotype information can be used to reduce vague alleles.
      - This option is specific to display methylation in allele specific mode.
-D    Minimum number of reads (depth) covered in this region or allele linked. 
      [default: 0|OFF]
-C    Minimum number of C (specified type) covered in this region or allele
      linked. [default: 0|OFF]
-W    Width of graphics reigon in inches. [default: 4]
-H    Height of graphics reigon in inches. [default: 4]
-R    Resolution in ppi. [default: 300]
      - only available for png device.</code></pre></li>
<li><p><strong>Example command</strong>:</p>
<p><code>cgmaptools tanghulu -r genome.fa -b WG.bam -l chr1:2000-2400 -t CG</code></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-14"></span>
<img src="CirclePlot.CG.chr1_3017150_3017200.png" alt="FragRegMC example" width="80%" />
<p class="caption">
Figure 7.5: FragRegMC example
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="coverage-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="other-ultilities.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"download": ["CGmapTools_documentation.pdf", "CGmapTools_documentation.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

</body>

</html>
