<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations.">
  <meta name="generator" content="bookdown 0.4 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  



<meta name="date" content="2017-09-17">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="coverage-analysis.html">
<link rel="next" href="other-ultilities.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<h3 style="text-align:center;font-family:'Times New Roman'">CGmapTools Documentation</h3>
<a href="https://cgmaptools.github.io"><h4 style="text-align:center"><- Back Home</h4></a>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="what-is-cgmaptools.html"><a href="what-is-cgmaptools.html"><i class="fa fa-check"></i><b>1</b> What is CGmapTools</a></li>
<li class="chapter" data-level="2" data-path="file-formats.html"><a href="file-formats.html"><i class="fa fa-check"></i><b>2</b> File Formats</a><ul>
<li class="chapter" data-level="2.1" data-path="file-formats.html"><a href="file-formats.html#atcgmap-format"><i class="fa fa-check"></i><b>2.1</b> ATCGmap Format</a></li>
<li class="chapter" data-level="2.2" data-path="file-formats.html"><a href="file-formats.html#cgmap-format"><i class="fa fa-check"></i><b>2.2</b> CGmap Format</a></li>
<li class="chapter" data-level="2.3" data-path="file-formats.html"><a href="file-formats.html#atcgbz-format"><i class="fa fa-check"></i><b>2.3</b> ATCGbz Format</a></li>
<li class="chapter" data-level="2.4" data-path="file-formats.html"><a href="file-formats.html#cgbz-format"><i class="fa fa-check"></i><b>2.4</b> CGbz Format</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="file-manipulation.html"><a href="file-manipulation.html"><i class="fa fa-check"></i><b>3</b> File Manipulation</a><ul>
<li class="chapter" data-level="3.1" data-path="file-manipulation.html"><a href="file-manipulation.html#convert"><i class="fa fa-check"></i><b>3.1</b> convert</a></li>
<li class="chapter" data-level="3.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch"><i class="fa fa-check"></i><b>3.2</b> fetch</a><ul>
<li class="chapter" data-level="3.2.1" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch-cgbz"><i class="fa fa-check"></i><b>3.2.1</b> fetch cgbz</a></li>
<li class="chapter" data-level="3.2.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch-atcgbz"><i class="fa fa-check"></i><b>3.2.2</b> fetch atcgbz</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="file-manipulation.html"><a href="file-manipulation.html#refill"><i class="fa fa-check"></i><b>3.3</b> refill</a></li>
<li class="chapter" data-level="3.4" data-path="file-manipulation.html"><a href="file-manipulation.html#intersect"><i class="fa fa-check"></i><b>3.4</b> intersect</a></li>
<li class="chapter" data-level="3.5" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2"><i class="fa fa-check"></i><b>3.5</b> merge2</a><ul>
<li class="chapter" data-level="3.5.1" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2-atcgmap"><i class="fa fa-check"></i><b>3.5.1</b> merge2 atcgmap</a></li>
<li class="chapter" data-level="3.5.2" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2-cgmap"><i class="fa fa-check"></i><b>3.5.2</b> merge2 cgmap</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist"><i class="fa fa-check"></i><b>3.6</b> mergelist</a><ul>
<li class="chapter" data-level="3.6.1" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist-tomatrix"><i class="fa fa-check"></i><b>3.6.1</b> mergelist tomatrix</a></li>
<li class="chapter" data-level="3.6.2" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist-tosingle"><i class="fa fa-check"></i><b>3.6.2</b> mergelist tosingle</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="file-manipulation.html"><a href="file-manipulation.html#sort"><i class="fa fa-check"></i><b>3.7</b> sort</a></li>
<li class="chapter" data-level="3.8" data-path="file-manipulation.html"><a href="file-manipulation.html#split"><i class="fa fa-check"></i><b>3.8</b> split</a></li>
<li class="chapter" data-level="3.9" data-path="file-manipulation.html"><a href="file-manipulation.html#select"><i class="fa fa-check"></i><b>3.9</b> select</a><ul>
<li class="chapter" data-level="3.9.1" data-path="file-manipulation.html"><a href="file-manipulation.html#select-region"><i class="fa fa-check"></i><b>3.9.1</b> select region</a></li>
<li class="chapter" data-level="3.9.2" data-path="file-manipulation.html"><a href="file-manipulation.html#select-site"><i class="fa fa-check"></i><b>3.9.2</b> select site</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snv-calling.html"><a href="snv-calling.html"><i class="fa fa-check"></i><b>4</b> SNV calling</a></li>
<li class="chapter" data-level="5" data-path="methylation-analysis.html"><a href="methylation-analysis.html"><i class="fa fa-check"></i><b>5</b> Methylation Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dms"><i class="fa fa-check"></i><b>5.1</b> dms</a></li>
<li class="chapter" data-level="5.2" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dmr"><i class="fa fa-check"></i><b>5.2</b> dmr</a></li>
<li class="chapter" data-level="5.3" data-path="methylation-analysis.html"><a href="methylation-analysis.html#asm"><i class="fa fa-check"></i><b>5.3</b> asm</a></li>
<li class="chapter" data-level="5.4" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbed"><i class="fa fa-check"></i><b>5.4</b> mbed</a></li>
<li class="chapter" data-level="5.5" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbin"><i class="fa fa-check"></i><b>5.5</b> mbin</a></li>
<li class="chapter" data-level="5.6" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mmbin"><i class="fa fa-check"></i><b>5.6</b> mmbin</a></li>
<li class="chapter" data-level="5.7" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mfg"><i class="fa fa-check"></i><b>5.7</b> mfg</a></li>
<li class="chapter" data-level="5.8" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mstat"><i class="fa fa-check"></i><b>5.8</b> mstat</a></li>
<li class="chapter" data-level="5.9" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mtr"><i class="fa fa-check"></i><b>5.9</b> mtr</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="coverage-analysis.html"><a href="coverage-analysis.html"><i class="fa fa-check"></i><b>6</b> Coverage Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac"><i class="fa fa-check"></i><b>6.1</b> oac</a><ul>
<li class="chapter" data-level="6.1.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac-bin"><i class="fa fa-check"></i><b>6.1.1</b> oac bin</a></li>
<li class="chapter" data-level="6.1.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac-stat"><i class="fa fa-check"></i><b>6.1.2</b> oac stat</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec"><i class="fa fa-check"></i><b>6.2</b> mec</a><ul>
<li class="chapter" data-level="6.2.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec-bin"><i class="fa fa-check"></i><b>6.2.1</b> mec bin</a></li>
<li class="chapter" data-level="6.2.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec-stat"><i class="fa fa-check"></i><b>6.2.2</b> mec stat</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>7</b> Graphics</a><ul>
<li class="chapter" data-level="7.1" data-path="graphics.html"><a href="graphics.html#lollipop"><i class="fa fa-check"></i><b>7.1</b> lollipop</a></li>
<li class="chapter" data-level="7.2" data-path="graphics.html"><a href="graphics.html#heatmap"><i class="fa fa-check"></i><b>7.2</b> heatmap</a></li>
<li class="chapter" data-level="7.3" data-path="graphics.html"><a href="graphics.html#fragreg"><i class="fa fa-check"></i><b>7.3</b> fragreg</a></li>
<li class="chapter" data-level="7.4" data-path="graphics.html"><a href="graphics.html#tanghulu"><i class="fa fa-check"></i><b>7.4</b> tanghulu</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="other-ultilities.html"><a href="other-ultilities.html"><i class="fa fa-check"></i><b>8</b> Other Ultilities</a><ul>
<li class="chapter" data-level="8.1" data-path="other-ultilities.html"><a href="other-ultilities.html#findccgg"><i class="fa fa-check"></i><b>8.1</b> findCCGG</a></li>
<li class="chapter" data-level="8.2" data-path="other-ultilities.html"><a href="other-ultilities.html#bed2fragreg"><i class="fa fa-check"></i><b>8.2</b> bed2fragreg</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="graphics" class="section level1">
<h1><span class="header-section-number">7</span> Graphics</h1>
<div id="lollipop" class="section level2">
<h2><span class="header-section-number">7.1</span> lollipop</h2>
<p>The consideration that we design novel <strong>Lollipop</strong> plot, is to be able to distinguish <strong>un-methylated sites</strong> and <strong>un-detected sites</strong>. Each covered cytosine would have a large round head; color and height of the bar represent DNA methylation level.</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> lollipop -h</code></pre></div>
<pre><code>#   Usage: cgmaptools lollipop [options] file
#         (aka mCLollipop) 
#   Description: Plot local mC level for multiple samples 
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2017-09-16 
#   Example: 
#       mCLollipop [-i input] -o gene.png 
#   -Input Format (-i)
#       Can be output by &quot;cgmaptools mergelist tomatrix&quot;. Use STDIN if omitted.
#       The 1st line (header line) is required.
#       Example: 
#          chr     pos     tag1    tag2    tag3
#          Chr1    111403  0.30    nan     0.80
#          Chr1    111406  0.66    0.40    0.60
#   -Site File (-s)
#       &gt;= 3 columns, the 1st line (header line) is required, using R color name or &quot;NaN&quot;. 
#       To show specific sites (such as DMS, SNV) at the bottom as triangles. 
#       Example: 
#           chr   pos       A_vs_B  B_vs_C  A_vs_C
#           chr1  13116801  NaN     NaN     darkgreen
#           chr1  13116899  NaN     red     NaN
#   -Region File (-b)
#       the first 4 columns are required.
#       To show specific region (such as DMR, Repeats) at the bottom as blocks. 
#       Example: 
#           chr1  213941196  213942363  hyper-DMR 
#           chr1  213942363  213943530  hypo-DMR 
#       #   chr   left       right      region-description
#   -annotation file (-a), refFlat Format:
#       To show the structure of genes/transcripts. One-line in annotation, one-track in figure. 
#       Example: 
#           GeneA   TransA  chr2  +      1000      2000       1100    1950     3     1100,1500,1700,  1200,1580,1950,
#       #   GeneID  TrandID ChrID Strand TransLeft TransRight CDSLeft CDSRight nExon ExonLefts        ExonRights
#       
#   
#   Options:
#       -i INFILE, --infile=INFILE
#           input file
#   
#       -a ANNOTATION, --annotation=ANNOTATION
#           [opt] annotation file name, refFlat format
#   
#       -o OUTFILE, --outfile=OUTFILE
#           [opt] output file
#   
#       -f FORMAT, --format=FORMAT
#           [opt] the format for output figure: pdf (default), png, eps
#   
#       -l LEFT, --left=LEFT
#           [opt] Left-most position
#   
#       -r RIGHT, --right=RIGHT
#           [opt] Right-most position
#   
#       -c CHR, --chr=CHR
#           [opt] chromosome name
#   
#       -s SITE, --site=SITE
#           [opt] file of site to be marked
#   
#       -b BED, --bed=BED
#           [opt] BED file for region to be markered
#   
#       -t TITLE, --title=TITLE
#           [opt] text shown on title
#   
#       -w WIDTH, --width=WIDTH
#           [opt] width (in inch). Default: 8.
#   
#       --height=HEIGHT
#           [opt] height (in inch). Default: 8.
#   
#       -h, --help
#           Show this help message and exit</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools lollipop -i matrix.CG.gz -a anno.refFlat -f pdf</code></p></li>
<li><p><strong>Figure examples</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-68"></span>
<img src="TRIM59.png" alt="Lollipop example-1" width="80%" />
<p class="caption">
Figure 7.1: Lollipop example-1
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-69"></span>
<img src="VCAN.png" alt="Lollipop example-2" width="80%" />
<p class="caption">
Figure 7.2: Lollipop example-2
</p>
</div>
<ul>
<li><p><strong>refFlat</strong> format</p></li>
<li><p>Example</p></li>
</ul>
<p><code>GeneA TransA  chr2    +   1000    2000    1100    1950    3   1100,1500,1700, 1200,1580,1950,</code></p>
<ul>
<li>Description</li>
</ul>
<blockquote>
<p>Col 1: Gene ID<br />
Col 2: Transcript ID<br />
Col 3: chromatine ID<br />
Col 4: strand, “+” or “-”<br />
Col 5: The left-most position of transcript<br />
Col 6: The right-most position of transcript<br />
Col 7: The left-most position of CDS<br />
Col 8: The right-most position of CDS<br />
Col 9: Number of exons<br />
Col 10: List of left-most position of exons, seperated by “,”<br />
Col 11: List of right-most position of exons, seperated by “,”</p>
</blockquote>
<ul>
<li>Convert GTF format to refFlat format</li>
</ul>
<p>The following is an example for Z. mays.</p>
<blockquote>
<p>“gtfToGenePred” is a command tool downloaded from UCSC utility.</p>
</blockquote>
<pre><code>gtfToGenePred  -genePredExt -geneNameAsName2 -allErrors  AGPv4.gtf AGPv4.GenePred
paste &lt;(cut -f13 AGPv4.GenePred) &lt;(cut -f1-10 AGPv4.GenePred) &gt; AGPv4.refFlat
paste &lt;(cut -f13 AGPv4.GenePred) &lt;(cut -f1-10 AGPv4.GenePred) | sed -i s/transcript://g | cut -f9 | gawk -F&quot;[\&quot;:]&quot; &#39;{print $3&quot;\t&quot;$6;}&#39; | sort -u &gt; trans_gene_ID
cut -f1-10 AGPv4.GenePred &gt; AGPv4.refFlat.tmp
gawk -F&quot;\t&quot; -vOFS=&quot;\t&quot; &#39;ARGIND==1{GeneID[$1]=$2;} ARGIND==2{printf GeneID[$1]&quot;\t&quot;$0}&#39; trans_gene_ID AGPv4.refFlat.tmp &gt; AGPv4.refFlat
rm ${GN}.refFlat.txt AGPv4.GenePred</code></pre>
</div>
<div id="heatmap" class="section level2">
<h2><span class="header-section-number">7.2</span> heatmap</h2>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> heatmap -h</code></pre></div>
<pre><code>#   Usage: cgmaptools heatmap [options]
#         (aka mCBinHeatmap)
#   Description: Plot methylation dynamics of target region for multiple samples [heatmap]
#   Contact:     Zhu, Ping; pingzhu.work@gmail.com
#   Last update: 2017-09-16
#   Example: 
#     mCBinHeatmap.R -i input -m white -o chr1.xxx-xxx.pdf 
#     -Input File Format: 
#     1st line is the header.
#     Each column contains methylation measurements of a sample. 
#     Example: 
#     Region  Sample1  Sample2 ...  
#     Region1 0.1      0.1     ...  
#     Region2 0.1      0.1     ...  
#   
#   
#   Options:
#       -i INFILE, --infile=INFILE
#           input file
#   
#       -o OUTFILE, --outfile=OUTFILE
#           [opt] output file name. [default: mCBinHeatmap.SysDate.pdf]
#   
#       -c, --cluster
#           [opt] cluster samples by methylation in regions. [default: FALSE]
#   
#       -l COLORLOW, --colorLow=COLORLOW
#           [opt] color used for the lowest methylation value. [default: cyan3]
#   
#       -m COLORMID, --colorMid=COLORMID
#           [opt] color used for the middle methylation value. [default: null]
#   
#       -b COLORHIGH, --colorHigh=COLORHIGH
#           [opt] color used for the highest methylation value. [default: coral2]
#   
#       -n COLORNUMBER, --colorNumber=COLORNUMBER
#           [opt] desired number of color elements in the panel. [default: 10]
#   
#       -W WIDTH, --width=WIDTH
#           [opt] width of figure (inch). [default: 7]
#   
#       -H HEIGHT, --height=HEIGHT
#           [opt] height of figure (inch). [default: 7]
#   
#       -f FORMAT, --format=FORMAT
#           [opt] format of output figure. Alternative: png. [default: pdf]
#   
#       -R RESOLUTION, --resolution=RESOLUTION
#           [opt] Resolution in ppi. Only available for png format. [default: 300]
#   
#       -h, --help
#           Show this help message and exit</code></pre>
<ul>
<li><p><strong>Example</strong>:</p>
<p><code>cgmaptools mmbin -l 1.CGmap,2.CGmap,3.CGmap &gt; mmbin.tab</code> <code>cgmaptools heatmap -i mmbin.tab -c -o cluster.pdf -f pdf</code></p></li>
<li><p><strong>Figure examples</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-71"></span>
<img src="mC_Cluster.png" alt="heatmap example-1" width="80%" />
<p class="caption">
Figure 7.3: heatmap example-1
</p>
</div>
</div>
<div id="fragreg" class="section level2">
<h2><span class="header-section-number">7.3</span> fragreg</h2>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> fragreg -h</code></pre></div>
<pre><code>#   Usage: cgmaptools fragreg [options]
#         (aka mCFragRegView) 
#   Description: Plot methylation dynamics of target and flanking region for multiple samples 
#   Contact:     Zhu, Ping; pingzhu.work@gmail.com
#   Last update: 2017-09-16
#   Example: 
#     FragRegView.R -i input -r 5 -o genebody.pdf 
#   -Input File Format: 
#     1st line is the header.
#   Each row contains methylation measurements of a sample. 
#   Example: 
#     Sample  Up1  Up2  ...  Region1  Region2 ...  Down1  Down2  ... 
#     Sample1 0.1  0.1  ...  0.2      0.2     ...  0.3    0.3    ...
#     Sample2 0.1  0.1  ...  0.2      0.2     ...  0.3    0.3    ...
#   
#   
#   Options:
#       -i INFILE, --infile=INFILE
#           input file
#   
#       -r RATIO, --ratio=RATIO
#           [opt] range ratio between target region and flanking region in plot. [default: 5]
#   
#       -o OUTFILE, --outfile=OUTFILE
#           [opt] output file name. [default: FragRegView.SysDate.pdf
#   
#       -W WIDTH, --width=WIDTH
#           [opt] width of figure (inch). [default: 7]
#   
#       -H HEIGHT, --height=HEIGHT
#           [opt] height of figure (inch). [default: 7]
#   
#       -f FORMAT, --format=FORMAT
#           [opt] format of output figure. Alternative: png. [default: pdf]
#   
#       -R RESOLUTION, --resolution=RESOLUTION
#           [opt] Resolution in ppi. Only available for png format. [default: 300]
#   
#       -h, --help
#           Show this help message and exit</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>for CHR in 1 2 3 4 5; do (for P in 1 2 3 4 5; do echo | gawk -vC=$CHR -vP=$P -vOFS=&quot;\t&quot; '{print &quot;chr&quot;C, P*1000, P*1000+1000, &quot;+&quot;;}' ; done) ; done | cgmaptools bed2fragreg -n 30 -F 50,50,50,50,50,50,50,50,50,50 -T 50,50,50,50,50,50,50,50,50,50 &gt; fragreg.bed</code> <code>gunzip -c WG.CGmap.gz | cgmaptools mfg -r fragreg.bed -c 2 -x CG &gt; WG.mfg</code></p></li>
<li><p><strong>Output format</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>Region_ID       R_1     R_2     R_3     R_4
sense_ave_mC    0.50    0.40    0.30    0.20
sense_sum_mC    5.0     4.0     3.0     2.0
sense_sum_NO    10      10      10      10   
anti_ave_mC     0.40    0.20    0.10    NaN
anti_sum_mC     8.0     4.0     2.0     0.0
anti_sum_NO     20      20      20      0
total_ave_mC    0.43    0.27    0.17    0.2
total_sum_mC    13.0    8.0     5.0     2.0
total_sum_NO    30      30      30      10</code></pre>
<ul>
<li><p><strong>Column Description</strong></p>
<ul>
<li>Tab-delimited file with header line</li>
<li>Content in 1st col is fixed</li>
<li>Column number is dynamic</li>
<li>Invalid number is annotated as “NaN”</li>
</ul></li>
</ul></li>
<li><p><strong>Output figure</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-73"></span>
<img src="GeneBody_mC_profile.png" alt="DNA methylation distribution across gene body" width="80%" />
<p class="caption">
Figure 7.4: DNA methylation distribution across gene body
</p>
</div>
</div>
<div id="tanghulu" class="section level2">
<h2><span class="header-section-number">7.4</span> tanghulu</h2>
<p>The <strong>Tanghulu</strong> plot is designed as show the methylation state on each cytosine by reads. (See what does “<em>Tanghulu</em>” strand for? <a href="https://en.wikipedia.org/wiki/Tanghulu">Wikipedia</a>)</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> tanghulu -h</code></pre></div>
<pre><code>#  DESCRIPTION
#          Circle plot representing DNA methylation of each C [defualt CpG] site
#          on each mapped reads.
#  
#  USAGE
#          cgmaptools tanghulu [options] -r &lt;ref&gt; -b &lt;bam&gt; -l chr1:133-144
#          or: cgmaptools tanghulu [options] -r &lt;ref&gt; -b &lt;bam&gt; -l chr1:133
#          (aka mCTanghulu)
#  
#          Options:
#          -r    Samtools indexed reference genome seqeunce, fasta format. eg. hg19.fa
#                - use samtools to index reference: samtools faidx &lt;hg19.fa&gt;
#          -b    Samtools indexed Bam file to view.
#                - use samtools to index bam file: samtools index &lt;input.bam&gt;
#          -l    Region in which to display DNA methylation.
#                - or specify a single position (eg. heterozygous SNP site), we will show allele specific methylation.
#          -s    Path to samtools eg. /home/user/bin/samtools
#                - by defualt, we try to search samtools in your system PATH.
#          -o    Output results to file [default: CirclePlot.Ctype.region.Date.pdf].
#          -t    C context. [default: CG]
#                - available context: C, CG, CH, CW, CC, CA, CT, CHG, CHH
#          -d    Ouput device. [default: pdf]
#                - alternative: png
#          -c    Seperate reads by chain. [default: OFF]
#                - specify this option to turn ON.
#          -v    Show vague allele linked reads. [ default: OFF]
#          -g    Genotype of heterozygous SNP site.
#                - This option provides two alleles of htSNP site. eg. AT
#                - The genotype information can be used to reduce vague alleles.
#                - This option is specific to display methylation in allele specific mode.
#          -D    Minimum number of reads (depth) covered in this region or allele linked. [default: 0|OFF]
#          -C    Minimum number of C (specified type) covered in this region or allele linked. [default: 0|OFF]
#          -W    Width of graphics reigon in inches. [default: 4]
#          -H    Height of graphics reigon in inches. [default: 4]
#          -R    Resolution in ppi. [default: 300]
#                - only available for png device.
#          -h    Help message.
#  
#  AUTHOR
#          Contact:     Zhu, Ping; pingzhu.work@gmail.com
#          Last update: 2016-12-07</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools tanghulu -r genome.fa -b WG.bam -l chr1:2000-2400 -t CG</code></p></li>
<li><p><strong>Output figure</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-75"></span>
<img src="CirclePlot.CG.chr1_3017150_3017200.png" alt="Tanghulu plot example" width="50%" />
<p class="caption">
Figure 7.5: Tanghulu plot example
</p>
</div>
<p>We also designed <strong>Tanghulu</strong> plot for visualizing reads that are support methylated, un-methylated, and vague reads for Allele-Specific Methylation (ASM) region.</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-76"></span>
<img src="TanghuluPlot.png" alt="Tanghulu plot show vague-reads" width="50%" />
<p class="caption">
Figure 7.6: Tanghulu plot show vague-reads
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="coverage-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="other-ultilities.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"download": ["CGmapTools_documentation.pdf", "CGmapTools_documentation.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
