<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations.">
  <meta name="generator" content="bookdown 0.4 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="A bisulfite sequencing analysis toolset with enhanced features on SNV calling and allele specific methylations and visualizations." />
  



<meta name="date" content="2017-09-17">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="snv-calling.html">
<link rel="next" href="coverage-analysis.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<h3 style="text-align:center;font-family:'Times New Roman'">CGmapTools Documentation</h3>
<a href="https://cgmaptools.github.io"><h4 style="text-align:center"><- Back Home</h4></a>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="what-is-cgmaptools.html"><a href="what-is-cgmaptools.html"><i class="fa fa-check"></i><b>1</b> What is CGmapTools</a></li>
<li class="chapter" data-level="2" data-path="file-formats.html"><a href="file-formats.html"><i class="fa fa-check"></i><b>2</b> File Formats</a><ul>
<li class="chapter" data-level="2.1" data-path="file-formats.html"><a href="file-formats.html#atcgmap-format"><i class="fa fa-check"></i><b>2.1</b> ATCGmap Format</a></li>
<li class="chapter" data-level="2.2" data-path="file-formats.html"><a href="file-formats.html#cgmap-format"><i class="fa fa-check"></i><b>2.2</b> CGmap Format</a></li>
<li class="chapter" data-level="2.3" data-path="file-formats.html"><a href="file-formats.html#atcgbz-format"><i class="fa fa-check"></i><b>2.3</b> ATCGbz Format</a></li>
<li class="chapter" data-level="2.4" data-path="file-formats.html"><a href="file-formats.html#cgbz-format"><i class="fa fa-check"></i><b>2.4</b> CGbz Format</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="file-manipulation.html"><a href="file-manipulation.html"><i class="fa fa-check"></i><b>3</b> File Manipulation</a><ul>
<li class="chapter" data-level="3.1" data-path="file-manipulation.html"><a href="file-manipulation.html#convert"><i class="fa fa-check"></i><b>3.1</b> convert</a></li>
<li class="chapter" data-level="3.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch"><i class="fa fa-check"></i><b>3.2</b> fetch</a><ul>
<li class="chapter" data-level="3.2.1" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch-cgbz"><i class="fa fa-check"></i><b>3.2.1</b> fetch cgbz</a></li>
<li class="chapter" data-level="3.2.2" data-path="file-manipulation.html"><a href="file-manipulation.html#fetch-atcgbz"><i class="fa fa-check"></i><b>3.2.2</b> fetch atcgbz</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="file-manipulation.html"><a href="file-manipulation.html#refill"><i class="fa fa-check"></i><b>3.3</b> refill</a></li>
<li class="chapter" data-level="3.4" data-path="file-manipulation.html"><a href="file-manipulation.html#intersect"><i class="fa fa-check"></i><b>3.4</b> intersect</a></li>
<li class="chapter" data-level="3.5" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2"><i class="fa fa-check"></i><b>3.5</b> merge2</a><ul>
<li class="chapter" data-level="3.5.1" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2-atcgmap"><i class="fa fa-check"></i><b>3.5.1</b> merge2 atcgmap</a></li>
<li class="chapter" data-level="3.5.2" data-path="file-manipulation.html"><a href="file-manipulation.html#merge2-cgmap"><i class="fa fa-check"></i><b>3.5.2</b> merge2 cgmap</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist"><i class="fa fa-check"></i><b>3.6</b> mergelist</a><ul>
<li class="chapter" data-level="3.6.1" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist-tomatrix"><i class="fa fa-check"></i><b>3.6.1</b> mergelist tomatrix</a></li>
<li class="chapter" data-level="3.6.2" data-path="file-manipulation.html"><a href="file-manipulation.html#mergelist-tosingle"><i class="fa fa-check"></i><b>3.6.2</b> mergelist tosingle</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="file-manipulation.html"><a href="file-manipulation.html#sort"><i class="fa fa-check"></i><b>3.7</b> sort</a></li>
<li class="chapter" data-level="3.8" data-path="file-manipulation.html"><a href="file-manipulation.html#split"><i class="fa fa-check"></i><b>3.8</b> split</a></li>
<li class="chapter" data-level="3.9" data-path="file-manipulation.html"><a href="file-manipulation.html#select"><i class="fa fa-check"></i><b>3.9</b> select</a><ul>
<li class="chapter" data-level="3.9.1" data-path="file-manipulation.html"><a href="file-manipulation.html#select-region"><i class="fa fa-check"></i><b>3.9.1</b> select region</a></li>
<li class="chapter" data-level="3.9.2" data-path="file-manipulation.html"><a href="file-manipulation.html#select-site"><i class="fa fa-check"></i><b>3.9.2</b> select site</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snv-calling.html"><a href="snv-calling.html"><i class="fa fa-check"></i><b>4</b> SNV calling</a></li>
<li class="chapter" data-level="5" data-path="methylation-analysis.html"><a href="methylation-analysis.html"><i class="fa fa-check"></i><b>5</b> Methylation Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dms"><i class="fa fa-check"></i><b>5.1</b> dms</a></li>
<li class="chapter" data-level="5.2" data-path="methylation-analysis.html"><a href="methylation-analysis.html#dmr"><i class="fa fa-check"></i><b>5.2</b> dmr</a></li>
<li class="chapter" data-level="5.3" data-path="methylation-analysis.html"><a href="methylation-analysis.html#asm"><i class="fa fa-check"></i><b>5.3</b> asm</a></li>
<li class="chapter" data-level="5.4" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbed"><i class="fa fa-check"></i><b>5.4</b> mbed</a></li>
<li class="chapter" data-level="5.5" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mbin"><i class="fa fa-check"></i><b>5.5</b> mbin</a></li>
<li class="chapter" data-level="5.6" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mmbin"><i class="fa fa-check"></i><b>5.6</b> mmbin</a></li>
<li class="chapter" data-level="5.7" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mfg"><i class="fa fa-check"></i><b>5.7</b> mfg</a></li>
<li class="chapter" data-level="5.8" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mstat"><i class="fa fa-check"></i><b>5.8</b> mstat</a></li>
<li class="chapter" data-level="5.9" data-path="methylation-analysis.html"><a href="methylation-analysis.html#mtr"><i class="fa fa-check"></i><b>5.9</b> mtr</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="coverage-analysis.html"><a href="coverage-analysis.html"><i class="fa fa-check"></i><b>6</b> Coverage Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac"><i class="fa fa-check"></i><b>6.1</b> oac</a><ul>
<li class="chapter" data-level="6.1.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac-bin"><i class="fa fa-check"></i><b>6.1.1</b> oac bin</a></li>
<li class="chapter" data-level="6.1.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#oac-stat"><i class="fa fa-check"></i><b>6.1.2</b> oac stat</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec"><i class="fa fa-check"></i><b>6.2</b> mec</a><ul>
<li class="chapter" data-level="6.2.1" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec-bin"><i class="fa fa-check"></i><b>6.2.1</b> mec bin</a></li>
<li class="chapter" data-level="6.2.2" data-path="coverage-analysis.html"><a href="coverage-analysis.html#mec-stat"><i class="fa fa-check"></i><b>6.2.2</b> mec stat</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>7</b> Graphics</a><ul>
<li class="chapter" data-level="7.1" data-path="graphics.html"><a href="graphics.html#lollipop"><i class="fa fa-check"></i><b>7.1</b> lollipop</a></li>
<li class="chapter" data-level="7.2" data-path="graphics.html"><a href="graphics.html#heatmap"><i class="fa fa-check"></i><b>7.2</b> heatmap</a></li>
<li class="chapter" data-level="7.3" data-path="graphics.html"><a href="graphics.html#fragreg"><i class="fa fa-check"></i><b>7.3</b> fragreg</a></li>
<li class="chapter" data-level="7.4" data-path="graphics.html"><a href="graphics.html#tanghulu"><i class="fa fa-check"></i><b>7.4</b> tanghulu</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="other-ultilities.html"><a href="other-ultilities.html"><i class="fa fa-check"></i><b>8</b> Other Ultilities</a><ul>
<li class="chapter" data-level="8.1" data-path="other-ultilities.html"><a href="other-ultilities.html#findccgg"><i class="fa fa-check"></i><b>8.1</b> findCCGG</a></li>
<li class="chapter" data-level="8.2" data-path="other-ultilities.html"><a href="other-ultilities.html#bed2fragreg"><i class="fa fa-check"></i><b>8.2</b> bed2fragreg</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="methylation-analysis" class="section level1">
<h1><span class="header-section-number">5</span> Methylation Analysis</h1>
<p>The <strong><em>CGmapTools</em></strong> supports both differentially methylated site (<strong>DMS</strong>) analyses and differentially methylated region (<strong>DMR</strong>) analyses are supported.</p>
<p>As the current available DNA methylome are either low coverage (such as WGBS) or fragmented in covered region (such as RRBS). In <strong><em>CGmapTools</em></strong>, we proposed a novel method <strong>dynamic fragmentation strategy</strong> for identifying DMRs between a pair of CGmap files.</p>
<div id="dms" class="section level2">
<h2><span class="header-section-number">5.1</span> dms</h2>
<p>Differentially methylated site analysis, supporting <strong><em>Chi-square</em></strong> and <strong><em>Fisher</em></strong> tests.</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> dms -h</code></pre></div>
<pre><code>#   Usage: cgmaptools dms [-i &lt;CGmapInter&gt;] [-m 5 -M 100] [-o output]
#         (aka CGmapInterDiffSite)
#   Description: 
#     Get the differentially methylated sites for two samples.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2017-01-20
#   Input Format, same as the output of CGmapIntersect.py:
#       Chr1  C  3541  CG  CG  0.8  4  5  0.4  4  10
#   Output Format:
#      chr1 C   4654    CG  CG  0.92    1.00    8.40e-01
#      chr1 C   4658    CHH CC  0.50    0.00    3.68e-04
#      chr1 G   8376    CG  CG  0.62    0.64    9.35e-01
#   
#   Options:
#     -h, --help            show this help message and exit
#     -i FILE               File name for CGmapInter, STDIN if omitted
#     -m INT, --min=INT     min coverage [default : 0]
#     -M INT, --max=INT     max coverage [default : 100]
#     -o OUTFILE            To standard output if omitted. Compressed output if
#                           end with .gz
#     -t STRING, --test-method=STRING
#                           chisq, fisher [default : chisq]</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools dms -i intersect_CG.gz -m 4 -M 100 -o DMS.gz -t fisher</code></p></li>
<li><p><strong>Output format</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>chr1    C   4654    CG  CG  0.92    1.00    8.40e-01
chr1    C   4658    CHH CC  0.50    0.00    3.68e-04
chr1    G   8376    CG  CG  0.62    0.64    9.35e-01</code></pre>
<ul>
<li><strong>Column Description</strong></li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-39"></span>
<img src="CGmapInterDiff.png" alt="Output format description for cgmaptools dms" width="100%" />
<p class="caption">
Figure 5.1: Output format description for cgmaptools dms
</p>
</div>
</div>
<div id="dmr" class="section level2">
<h2><span class="header-section-number">5.2</span> dmr</h2>
<p>Differentially methylated region analysis, using <strong>dynamic fragmentation strategy</strong> .</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> dmr -h</code></pre></div>
<pre><code>#   Usage: cgmaptools dmr [-i &lt;CGmapInter&gt;] [-m 5 -M 100] [-o output]
#         (aka CGmapInterDiffReg)
#   Description: 
#     Get the differentially methylated sites by Fisher&#39;s exact test.
#   Author:       Guo, Weilong; guoweilong@126.com; 
#   Last Updated: 2017-08-12
#   Input Format, same as the output of CGmapIntersect.py:
#      chr1  C  3541  CG  CG  0.8  4  5  0.4  4  10
#   Output Format, Ex:
#     #chr      start   end     t       pv          mC_A    mC_B    N_site
#      chr1 1004572 1004574 inf     0.00e+00    0.1100  0.0000  20
#      chr1 1009552 1009566 -0.2774 8.08e-01    0.0200  0.0300  15
#      chr1 1063405 1063498 0.1435  8.93e-01    0.6333  0.5733  5
#   
#   
#   Options:
#     -h, --help            show this help message and exit
#     -i FILE               File name for CGmapInter, STDIN if omitted
#     -c INT, --minCov=INT  min coverage [default : 4]
#     -C INT, --maxCov=INT  max coverage [default : 500]
#     -s INT, --minStep=INT
#                           min step in bp [default : 100]
#     -S INT, --maxStep=INT
#                           max step in bp [default : 1000]
#     -n INT, --minNSite=INT
#                           min N sites [default : 5]
#     -o OUTFILE            To standard output if omitted. Compressed output if
#                           end with .gz</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools dmr -i intersect_CG.gz -o DMR.gz</code></p></li>
<li><p><strong>Output format</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>chr1    1004572 1004574 inf     0.00e+00    0.1100  0.0000
chr1    1009552 1009566 -0.2774 8.08e-01    0.0200  0.0300
chr1    1063405 1063498 0.1435  8.93e-01    0.6333  0.5733
chr1    1082130 1082133 -0.0822 9.42e-01    0.5000  0.5550
chr1    1123931 1123933 inf     0.00e+00    0.0600  0.0000</code></pre>
<ul>
<li><strong>Column Description</strong></li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-41"></span>
<img src="CGmapIDR.png" alt="Output format description for cgmaptools dmr" width="100%" />
<p class="caption">
Figure 5.2: Output format description for cgmaptools dmr
</p>
</div>
<ul>
<li><strong>Dynamic Fragment Strategy</strong></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-42"></span>
<img src="Dynamic_Fragment_Strategy.png" alt="Dynamic Fragmentation Strategy" width="100%" />
<p class="caption">
Figure 5.3: Dynamic Fragmentation Strategy
</p>
</div>
</div>
<div id="asm" class="section level2">
<h2><span class="header-section-number">5.3</span> asm</h2>
<p>Feeding with the precisely predicted heterozygous SNVs (by <code>cgmaptools snv</code>), <strong><em>CGmapTools</em></strong> can identify Allele-Specific Methylated (<strong>ASM</strong>) regions from BAM files.</p>
<p>Following showed an interesting ASM region by analysing a previous cohort (<a href="http://www.nature.com/articles/srep32207">Weilong Guo, et al., Scientifc Report, 2016</a>).</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-43"></span>
<img src="ASM_ex.png" alt="Examples showed allele-specific methylated region reported by CGmapTools" width="50%" />
<p class="caption">
Figure 5.4: Examples showed allele-specific methylated region reported by CGmapTools
</p>
</div>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> asm -h</code></pre></div>
<pre><code>#   DESCRIPTION
#           Allele specific methylated region/site calling
#           * Fisher exact test for site calling.
#           * Students&#39; t-test for region calling.
#   
#   USAGE
#           cgmaptools asm [options] -r &lt;ref.fa&gt; -b &lt;input.bam&gt; -l &lt;snp.vcf&gt;
#           (aka ASM)
#   
#           Options:
#           -r    Samtools indexed reference genome seqeunce, fasta format. eg. hg19.fa
#                 - use samtools to index reference first: samtools faidx hg19.fa
#           -b    Samtools indexed Bam format file.
#                 - use samtools to index bam file first: samtools index &lt;input.bam&gt;
#           -l    SNPs in vcf file format.
#           -s    Path to samtools eg. /home/user/bin/samtools
#                 - by defualt, we try to search samtools in your system PATH,
#           -o    Output results to file. [default: STDOUT]
#           -t    C context. [default: CG]
#                 - available context: C, CG, CH, CW, CC, CA, CT, CHG, CHH
#           -m    Specify calling mode. [default: asr]
#                 - alternative: ass
#                 - asr: allele specific methylated region
#                 - ass: allele specific methylated site
#           -d    Minimum number of read for each allele linked site to call ass. [default: 3]
#                 - ass specific.
#           -n    Minimum number of C site each allele linked to call asr. [default: 2]
#                 - asr specific.
#           -D    Minimum read depth for C site to call methylation level when calling asr. [default: 1]
#                 - asr specific.
#           -L    Low methylation level threshold. [default: 0.2]
#                 - allele linked region [or site] with low methylation level should be no greater than this threshold.
#           -H    High methylation level threshold. [default: 0.8]
#                 - allele linked region[or site] with high methylation level should be no less than this threshold.
#           -q    Adjusted p value using Benjamini &amp; Hochberg (1995) (&quot;BH&quot; or its alias &quot;fdr&quot;). [default: 0.05]
#           -h    Help message.
#   
#   AUTHOR
#           Contact:     Zhu, Ping; pingzhu.work@gmail.com
#           Last update: 2016-12-07</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>gawk '{if(/^#/){print}else{print &quot;chr&quot;$0;}}' bayes.vcf &gt; bayes2.vcf</code></p>
<p><code>cgmaptools asm -r genome.fa -b WG.bam -l bayes2.vcf &gt; WG.asm</code></p></li>
<li><p><strong>Output format for ASS (Allele-Specific methylated Site)</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>Chr    SNP_Pos  Ref    Allele1     Allele2      C_Pos   Allele1_linked_C    Allele2_linked_C    Allele1_linked_C_met    Allele2_linked_C_met    pvalue   fdr    ASM
Chr1    8949221 T         T             A           8949252 30,2    6,0 0.94    1.00          1.00e+00         1.00e+00      FALSE
Chr1    8965481 A         A            T               8965494  12,3    12,4    0.80    0.75          1.00e+00         1.00e+00      FALSE</code></pre>
<ul>
<li><strong>Column Description</strong></li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-45"></span>
<img src="TAB_ASS.png" alt="Output format description for cgmaptools asm -m ass" width="100%" />
<p class="caption">
Figure 5.5: Output format description for cgmaptools asm -m ass
</p>
</div>
<ul>
<li><p><strong>Output format for ASR (Allele-Specific methylated Region)</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>Chr        Pos          Ref   Allele1   Allele2       Allele1_linked_C   Allele2_linked_C  Allele1_linked_C_met   Allele2_linked_C_met   pvalue      fdr             ASM
chr1       8943402   A        A        T                      1-1                         0.8-1                        1              0.9                        5.01e-01   7.52e-01    FALSE
chr1       8966879   C        C        G                    0.93-0-0     0.81-0-0         0.31            0.27                       9.27e-01    9.53e-01   FALSE</code></pre>
<ul>
<li><strong>Column Description</strong></li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-46"></span>
<img src="TAB_ASR.png" alt="Output format description for cgmaptools asm -m asr" width="80%" />
<p class="caption">
Figure 5.6: Output format description for cgmaptools asm -m asr
</p>
</div>
</div>
<div id="mbed" class="section level2">
<h2><span class="header-section-number">5.4</span> mbed</h2>
<p>The <code>cgmaptools mbed</code> command will calculated <strong>one</strong> DNA methylation level for <strong>all the investiaged regions</strong>, which is different from <code>cgmaptools mtr</code>.</p>
<p>For example, this function can be applied when calculating the average DNA methylation levels in regions, such as promoter, gene body, specific Transposon Elements (TEs).</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> mbed -h</code></pre></div>
<pre><code>#   Usage:  cgmaptools mbed [-i &lt;CGmap&gt;]  -b &lt;regin.bed&gt; [-c 5 -C 500 -s] 
#         (aka CGmapMethylInBed)
#   Description: Calculated bulk average methylation levels in given regions.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2017-01-20
#   Options:
#      -i  String, CGmap file; use STDIN if not specified
#          Please use &quot;gunzip -c &lt;input&gt;.gz &quot; and pipe as input for gzipped file.
#          Ex: chr1 G   3000851 CHH CC  0.1 1   10
#      -b  String, BED file, should have at least 4 columns
#          Ex: chr1 3000000 3005000 -
#      -c  Int, minimum Coverage [Default: 5] 
#      -C  Int, maximum Coverage [Default: 500] 
#      -s  Strands would be distinguished when specified
#      -h  help
#   
#   Output to STDOUT:
#       Title         Count    mean_mC
#       sense         34       0.2353
#       antisense     54       0.2778
#       total         88       0.2614
#   Notice:
#       The overlapping of regions would not be checked. 
#       A site might be considered multiple times.</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>zcat WG.CGmap.gz | cgmaptools mbed -b region.bed</code></p></li>
<li><p><strong>Output format</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>chr   sense_Count   sense_mC    anti_Count  anti_mC all_Count   all_mC
chr1    203   0.08127     178     0.1148      381     0.09692
chr2    185   0.07045     257     0.05586     442     0.06197
chr3    313   0.1042      250     0.1358      563     0.1182
chr4    300   0.1218      271     0.13      571   0.1257
chr5    282   0.1272      222     0.1589      504     0.1412</code></pre></li>
</ul>
</div>
<div id="mbin" class="section level2">
<h2><span class="header-section-number">5.5</span> mbin</h2>
<p>This function will calculated the average methylation levels in equal-length bins, across genome, generating both summary table and distribution graph.</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-48"></span>
<img src="mbin_ex.png" alt="Output figure example for cgmaptools mbin" width="100%" />
<p class="caption">
Figure 5.7: Output figure example for cgmaptools mbin
</p>
</div>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> mbin -h</code></pre></div>
<pre><code>#   Usage: cgmaptools mbin [-i &lt;CGmap&gt;] [-c 10 --CXY 5 -B 5000000] 
#         (aka CGmapMethInBins)
#   Description: Generate the methylation in Bins.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2016-10-26
#   Output Ex:
#      chr1    1       5000    0.0000
#      chr1    5001    10000   0.0396
#      chr2    1       5000    0.0755
#      chr2    5001    10000   0.0027
#      chr3    1       5000    na
#   
#   Options:
#     -h, --help            show this help message and exit
#     -i FILE               File name end with .CGmap or .CGmap.gz. If not
#                           specified, STDIN will be used.
#     -B BIN_SIZE           Define the size of bins [Default: 5000000]
#     -c COVERAGE           The minimum coverage for site selection [Default: 10]
#     -C CONTEXT, --context=CONTEXT
#                           specific context: CG, CH, CHG, CHH, CA, CC, CT, CW
#                           use all sites if not specified
#     --cXY=COVERAGEXY      Coverage for chrX/Y should be half that of autosome
#                           for male [Default: same with -c]
#     -f FIGTYPE, --figure-type=FIGTYPE
#                           png, pdf, eps. Will not generate figure if not
#                           specified
#     -H FLOAT              Height of figure in inch [Default: 4]
#     -W FLOAT              Width of figure in inch [Default: 8]
#     -p STRING             Prefix for output figures
#     -t STRING, --title=STRING
#                           title in the output figures</code></pre>
<ul>
<li><strong>Example</strong></li>
</ul>
<p><code>cgmaptools mbin -i WG.CGmap.gz -B 500 -c 4 -f png -t WG -p WG &gt; mbin.WG.data</code></p>
<ul>
<li><p><strong>File format</strong></p>
<p>The output format:</p>
<pre><code>chr1    1       5000    0.0000
chr1    5001    10000   0.0396
chr2    1       5000    0.0755
chr2    5001    10000   0.0027
chr3    1       5000    na</code></pre></li>
</ul>
</div>
<div id="mmbin" class="section level2">
<h2><span class="header-section-number">5.6</span> mmbin</h2>
<p>This function will calculate the average methylation levels in equal-length bins for <strong>mulitple</strong> samples, generating a summary table.</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> mmbin -h</code></pre></div>
<pre><code>#   Usage: cgmaptools mmbin [-l &lt;1.CGmap[,2.CGmap,..]&gt;] [-c 10 --CXY 5 -B 5000000] 
#         (aka CGmapsMethInBins)
#   Description: Generate the methylation in Bins.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2016-12-07
#   Output Ex:
#      chr1    1       5000    0.0000
#      chr1    5001    10000   0.0396
#      chr2    1       5000    0.0755
#      chr2    5001    10000   0.0027
#      chr3    1       5000    na
#   
#   Options:
#     -h, --help            show this help message and exit
#     -l FILE               File name list, end with .CGmap or .CGmap.gz. If not
#                           specified, STDIN will be used.
#     -t FILE               List of samples
#     -B BIN_SIZE           Define the size of bins [Default: 5000000]
#     -C CONTEXT, --context=CONTEXT
#                           specific context: CG, CH, CHG, CHH, CA, CC, CT, CW
#                           use all sites if not specified
#     -c COVERAGE           The minimum coverage for site selection [Default: 10]
#     --cXY=COVERAGEXY      Coverage for chrX/Y should be half that of autosome
#                           for male [Default: same with -c]</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools mmbin -l WG.CGmap.gz,RR.CGmap.gz,RR2.CGmap.gz,merge.CGmap.gz -c 4 -B 2000 | gawk '{printf(&quot;%s:%s-%s&quot;, $1, $2, $3); for(i=4;i&lt;=NF;i++){printf(&quot;\t%s&quot;, $i);} printf(&quot;\n&quot;);}' &gt; mmbin</code></p>
<p>```</p></li>
<li><p><strong>Output format</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>chr     pos1        pos2      tag1    tag2    tag3
Chr1    111403    113403    0.05    nan     0.02
Chr1    111500      112500  1.00    0.80    0.60
Chr2    20000       20500   0.96    0.33    0.66</code></pre>
<ul>
<li><strong>Column Description</strong></li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-51"></span>
<img src="TAB_RMM.png" alt="Output format description for cgmaptools mmbin" width="100%" />
<p class="caption">
Figure 5.8: Output format description for cgmaptools mmbin
</p>
</div>
</div>
<div id="mfg" class="section level2">
<h2><span class="header-section-number">5.7</span> mfg</h2>
<p>The <code>cgmaptools mfg</code> supports studys of Methylation in FraGmented regions. The function can be applied to draw mC distribution across gene body, transposon elements, and other user-provided regions. The fragmented regions can be generated using <code>cgmaptools bed2fragreg</code>.</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> mfg -h</code></pre></div>
<pre><code>#   Usage:  cgmaptools mfg [-i &lt;CGmap&gt;]  -r &lt;region&gt; [-c 5 -C 500] 
#   Description: Calculated methylation profile across fragmented regions.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2017-01-20
#   Options:
#      -i  String, CGmap file; use STDIN if not specified
#          Please use &quot;gunzip -c &lt;input&gt;.gz &quot; and pipe as input for gzipped file.
#          chr1 G   851 CHH CC  0.1 1   10
#      -r  String, Region file, at least 4 columns
#          Format: chr  strand  pos_1   pos_2   pos_3   ...
#          Regions would be considered as [pos_1, pos_2), [pos_2, pos_3)
#          Strand information will be used for distinguish sense/antisense strand
#          Ex:
#          #chr strand  U1  R1  R2  D1  End
#          chr1 +   600 700 800 900 950
#          chr1 -   1600    1500    1400    1300    1250
#      -c  Int, minimum Coverage [Default: 5] 
#      -C  Int, maximum Coverage [Default: 500] 
#          Sites exceed the coverage range will be discarded
#      -x  String, context [use all sites by default] 
#          string can be CG, CH, CHG, CHH, CA, CC, CT, CW
#      -h  help
#   Output to STDOUT:
#      Region_ID       U1      R1      R2      D1
#      sense_ave_mC    0.50    0.40    0.30    0.20
#      sense_sum_mC    5.0     4.0     3.0     2.0
#      sense_sum_NO    10      10      10      10
#      anti_ave_mC     0.40    0.20    0.10    NaN
#      anti_sum_mC     8.0     4.0     2.0     0.0
#      anti_sum_NO     20      20      20      0
#      total_ave_mC    0.43    0.27    0.17    0.2
#      total_sum_mC    13.0    8.0     5.0     2.0
#      total_sum_NO    30      30      30      10</code></pre>
<ul>
<li><p><strong>Example</strong>:</p>
<p><code>for CHR in 1 2 3 4 5; do (for P in 1 2 3 4 5; do echo | gawk -vC=$CHR -vP=$P -vOFS=&quot;\t&quot; '{print &quot;chr&quot;C, P*1000, P*1000+1000, &quot;+&quot;;}' ; done) ; done | cgmaptools bed2fragreg -n 30 -F 50,50,50,50,50,50,50,50,50,50 -T 50,50,50,50,50,50,50,50,50,50 &gt; fragreg.bed</code></p>
<p><code>gunzip -c WG.CGmap.gz | cgmaptools mfg -r fragreg.bed -c 2 -x CG &gt; WG.mfg</code></p>
<p><code>cgmaptools fragreg -i WG.mfg -f pdf -o WG_mfg.pdf</code></p></li>
</ul>
</div>
<div id="mstat" class="section level2">
<h2><span class="header-section-number">5.8</span> mstat</h2>
<p><strong><em>CGmapTools</em></strong> provide resourceful statistic analysis on DNA methylation globally. The <code>cgmaptools mstat</code> command will generated a table summary, together with several graphs:</p>
<ul>
<li>mC contributions of different contexts in Pie chart</li>
<li>Bulk mC levels of different contexts</li>
<li>Fragmented distribution of mC in different contexts</li>
</ul>
<p>The methylation contexts are diffferent for plants and animals. - For plants, the contexts for DNA methylations are known as <strong>CG</strong>, <strong>CHG</strong> and <strong>CHH</strong>, where <span class="math inline">\(H=\{A,C,T\}\)</span>. - For animals, the situation is different. In 2014, <a href="http://nar.oxfordjournals.org/content/42/5/3009.abstract">Weilong Guo, et al.</a> showed that it is unnecessary to seperate CHG methylations and CHH methylations in human. In 2016, <a href="http://www.nature.com/articles/srep32207">Weilong Guo, et al.</a> designed <strong><em>MiDD</em></strong> method and de novo predicted the main separated contexts for non-CG (CH) methylation should be CW (<span class="math inline">\(W=\{A,T\}\)</span>) and CC, and mCW is cell-type specific and conserved between human and mice. In <strong><em>CGmapTools</em></strong>, we support both human view (<strong>CG</strong>, <strong>CW</strong> and <strong>CC</strong>) and plant view (<strong>CG</strong>, <strong>CHG</strong>, and <strong>CHH</strong>) for DNA methylation contexts.</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> mstat -h</code></pre></div>
<pre><code>#   Usage: cgmaptools mstat [-i &lt;CGmap&gt;]  
#         (aka CGmapStatMeth)
#   Description: Generate the bulk methylation.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2016-12-08
#   Output Ex:
#      MethStat        context C       CG      CHG      CHH     CA      CC      CT      CH      CW
#      mean_mC         global  0.0798  0.3719  0.0465  0.0403  0.0891  0.0071  0.0241  0.0419  0.0559
#      sd_mCbyChr      global  0.0078  0.0341  0.0163  0.0110  0.0252  0.0049  0.0076  0.0096  0.0148
#      count_C         global  10000   1147    2332    6521    3090    2539    3224    8853    6314
#      contrib_mC      global  1.0000  0.5348  0.1360  0.3292  0.3452  0.0228  0.0973  0.4652  0.4424
#      quant_mC        [0]     8266    471     2012    5783    2422    2421    2952    7795    5374
#      quant_mC   (0.00 ,0.20] 705     182     155     368     272     97      154     523     426
#      mean_mC_byChr   chr1    0.0840  0.4181  0.0340  0.0412  0.0794  0.0065  0.0251  0.0393  0.0513
#      mean_mC_byChr   chr10   0.0917  0.4106  0.0758  0.0421  0.0968  0.0097  0.0349  0.0502  0.0655
#   
#   Options:
#     -h, --help            show this help message and exit
#     -i FILE               File name end with .CGmap or .CGmap.gz. If not
#                           specified, STDIN will be used.
#     -c COVERAGE           The minimum coverage for site selection [Default: 10]
#     -f FILE, --figure-type=FILE
#                           png, pdf, eps. Will not generate figure if not
#                           specified
#     -H FLOAT              Height of figure in inch [Default: 3]
#     -W FLOAT              Width of figure in inch [Default: 8]
#     -p STRING             Prefix for output figures
#     -t STRING, --title=STRING
#                           title in the output figures</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools mstat -i WG.CGmap.gz -c 4 -f png -p WG -t WG &gt; WG.mstat.data</code></p></li>
<li><p><strong>File format</strong></p>
<p>The output format:</p>
<pre><code>MethStat        context C       CG      CHG     CHH     CA      CC      CT      CH      CW
mean_mC         global  0.0798  0.3719  0.0465  0.0403  0.0891  0.0071  0.0241  0.0419  0.0559
sd_mCbyChr      global  0.0078  0.0341  0.0163  0.0110  0.0252  0.0049  0.0076  0.0096  0.0148
count_C         global  10000   1147    2332    6521    3090    2539    3224    8853    6314
contrib_mC      global  1.0000  0.5348  0.1360  0.3292  0.3452  0.0228  0.0973  0.4652  0.4424
quant_mC        [0]     8266    471     2012    5783    2422    2421    2952    7795    5374
quant_mC   (0.00 ,0.20] 705     182     155     368     272     97      154     523     426
mean_mC_byChr   chr1    0.0840  0.4181  0.0340  0.0412  0.0794  0.0065  0.0251  0.0393  0.0513
mean_mC_byChr   chr10   0.0917  0.4106  0.0758  0.0421  0.0968  0.0097  0.0349  0.0502  0.0655</code></pre></li>
<li><p><strong>Output figures</strong></p></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-54"></span>
<img src="Pie_mC_contribution.png" alt="mC contribution example" width="80%" />
<p class="caption">
Figure 5.9: mC contribution example
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-55"></span>
<img src="DNAmethylationLevel_ByContext.png" alt="Bulk mC example" width="80%" />
<p class="caption">
Figure 5.10: Bulk mC example
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-56"></span>
<img src="DNAmethylationDistribution_ByContext.png" alt="mC fragmented distribution example" width="80%" />
<p class="caption">
Figure 5.11: mC fragmented distribution example
</p>
</div>
</div>
<div id="mtr" class="section level2">
<h2><span class="header-section-number">5.9</span> mtr</h2>
<p>The <code>cgmaptools mtr</code> command will calculated the DNA methylation levels for each investiaged region.</p>
<ul>
<li><strong>Command</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">cgmaptools</span> mtr -h</code></pre></div>
<pre><code>#   Usage: cgmaptools mtr [-i &lt;CGmap&gt;] -r &lt;region&gt; [-o &lt;output&gt;]
#         (aka CGmapToRegion)
#   Description: Calculated the methylation levels in regions in two ways.
#   Contact:     Guo, Weilong; guoweilong@126.com
#   Last Update: 2017-01-20
#   Format of Region file:
#     #chr    start_pos  end_pos
#      chr1   8275       8429
#   Output file format:
#      #chr  start_pos  end_pos  mean(mC)  #_C  #read(C)/#read(T+C)  #read(T+C)
#      chr1   8275       8429     0.34     72         0.40             164
#   Note: The two input CGmap files should be sorted by Sort_chr_pos.py first.
#         This script would not distinguish CG/CHG/CHH contexts.
#   
#   Options:
#     -h, --help  show this help message and exit
#     -i FILE     File name end with .CGmap or .CGmap.gz. If not specified, STDIN
#                 will be used.
#     -r FILE     Filename for region file, support *.gz
#     -o OUTFILE  To standard output if not specified.</code></pre>
<ul>
<li><p><strong>Example</strong></p>
<p><code>cgmaptools mtr -i WG.CGmap.gz -r region.bed -o WG.mtr.gz</code></p></li>
<li><p><strong>Input region format</strong></p>
<pre><code>#chr    start_pos  end_pos
chr1    8275       8429</code></pre></li>
<li><p><strong>Output format</strong></p>
<ul>
<li><strong>Example</strong></li>
</ul>
<pre><code>chr1   8275   8429    0.34   72    0.40   164
chr1   8899   8999    0.20   40    0.33   198
chr2   8275   8429    0.50   12    0.45   40</code></pre>
<ul>
<li><strong>Column Description</strong></li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-58"></span>
<img src="TAB_RSM.png" alt="Output format description for cgmaptools mtr" width="100%" />
<p class="caption">
Figure 5.12: Output format description for cgmaptools mtr
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="snv-calling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="coverage-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"download": ["CGmapTools_documentation.pdf", "CGmapTools_documentation.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
